// @tnwx/commons/dist/entity/msg/in/InAuthEvent.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const BaseMsg_1 = require("../BaseMsg");
/**
 * @author Javen
 * @copyright javendev@126.com
 * @description 企业开放平台授权通知事件
 */
class InAuthEvent extends BaseMsg_1.BaseMsg {
    constructor(suiteId, infoType, timeStamp, authCode, authCorpId) {
        super();
        this.suiteid = suiteId;
        this.infotype = infoType;
        this.timestamp = timeStamp;
        this.authcode = authCode;
        this.authcorpid = authCorpId;
    }
    get suiteId() {
        return this.suiteid;
    }
    set suiteId(suiteId) {
        this.suiteid = suiteId;
    }
    get infoType() {
        return this.infotype;
    }
    set infoType(infoType) {
        this.infotype = infoType;
    }
    get timeStamp() {
        return this.timestamp;
    }
    set timeStamp(timeStamp) {
        this.timestamp = timeStamp;
    }
    get authCode() {
        return this.authcode;
    }
    set authCode(authCode) {
        this.authcode = authCode;
    }
    get authCorpId() {
        return this.authcorpid;
    }
    set authCorpId(authCorpId) {
        this.authcorpid = authCorpId;
    }
}
exports.InAuthEvent = InAuthEvent;
InAuthEvent.CREATE_AUTH = 'create_auth';
InAuthEvent.CHANGE_AUTH = 'change_auth';
InAuthEvent.CANCEL_AUTH = 'cancel_auth';
