// @tnwx/commons/dist/entity/msg/in/InAuthMpEvent.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const BaseMsg_1 = require("../BaseMsg");
/**
 * @author Javen
 * @copyright javendev@126.com
 * @description 微信开放平台授权通知事件
 */
class InAuthMpEvent extends BaseMsg_1.BaseMsg {
    constructor(appId, infoType, createTime, authAppId, authCode, authCodeExpiredTime, preAuthCode) {
        super();
        this.appid = appId;
        this.infotype = infoType;
        this.createtime = createTime;
        this.authappid = authAppId;
        this.authcode = authCode;
        this.authcodeexpiredtime = authCodeExpiredTime;
        this.preauthcode = preAuthCode;
    }
    get appId() {
        return this.appid;
    }
    set appId(appId) {
        this.appid = appId;
    }
    get infoType() {
        return this.infotype;
    }
    set infoType(infoType) {
        this.infotype = infoType;
    }
    get createTime() {
        return this.createtime;
    }
    set timeStamp(createTime) {
        this.createtime = createTime;
    }
    get authAppId() {
        return this.authappid;
    }
    set authAppId(authAppId) {
        this.authappid = authAppId;
    }
    get authCode() {
        return this.authcode;
    }
    set authCode(authCode) {
        this.authcode = authCode;
    }
    get authCodeExpiredTime() {
        return this.authcodeexpiredtime;
    }
    set authCodeExpiredTime(authCodeExpiredTime) {
        this.authcodeexpiredtime = authCodeExpiredTime;
    }
    get preAuthCode() {
        return this.preauthcode;
    }
    set preAuthCode(preAuthCode) {
        this.preauthcode = preAuthCode;
    }
}
exports.InAuthMpEvent = InAuthMpEvent;
InAuthMpEvent.CREATE_AUTH = 'authorized';
InAuthMpEvent.CHANGE_AUTH = 'updateauthorized';
InAuthMpEvent.CANCEL_AUTH = 'unauthorized';
