// @tnwx/commons/dist/entity/msg/in/InBatchJobResult.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const BaseMsg_1 = require("../BaseMsg");
/**
 * @author Javen
 * @copyright javendev@126.com
 * @description 异步任务回调通知
 */
class InBatchJobResult extends BaseMsg_1.BaseMsg {
    constructor(serviceCorpId, infoType, timeStamp, authCorpId, jobId, jobType) {
        super();
        this.servicecorpid = serviceCorpId;
        this.infotype = infoType;
        this.timestamp = timeStamp;
        this.authcorpid = authCorpId;
        this.jobid = jobId;
        this.jobtype = jobType;
    }
    get serviceCorpId() {
        return this.servicecorpid;
    }
    set serviceCorpId(serviceCorpId) {
        this.servicecorpid = serviceCorpId;
    }
    get infoType() {
        return this.infotype;
    }
    set infoType(infoType) {
        this.infotype = infoType;
    }
    get timeStamp() {
        return this.timestamp;
    }
    set timeStamp(timeStamp) {
        this.timestamp = timeStamp;
    }
    get authCorpId() {
        return this.authcorpid;
    }
    set authCorpId(authCorpId) {
        this.authCorpId = authCorpId;
    }
    get jobId() {
        return this.jobid;
    }
    set jobId(jobId) {
        this.jobid = jobId;
    }
    get jobType() {
        return this.jobtype;
    }
    set jobType(jobType) {
        this.jobtype = jobType;
    }
}
exports.InBatchJobResult = InBatchJobResult;
InBatchJobResult.INFO_TYPE = 'batch_job_result';
