// @tnwx/commons/dist/entity/msg/in/InComponentVerifyTicket.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const BaseMsg_1 = require("../BaseMsg");
/**
 * @author Javen
 * @copyright javendev@126.com
 * @description 推送 component_verify_ticket
 */
class InComponentVerifyTicket extends BaseMsg_1.BaseMsg {
    constructor(appId, infoType, createTime, ticket) {
        super();
        this.appid = appId;
        this.infotype = infoType;
        this.createtime = createTime;
        this.ticket = ticket;
    }
    get appId() {
        return this.appid;
    }
    set appId(appId) {
        this.appid = appId;
    }
    get infoType() {
        return this.infotype;
    }
    set infoType(infoType) {
        this.infotype = infoType;
    }
    get createTime() {
        return this.createtime;
    }
    set timeStamp(createTime) {
        this.createtime = createTime;
    }
    get getTicket() {
        return this.ticket;
    }
    set setTicket(ticket) {
        this.ticket = ticket;
    }
}
exports.InComponentVerifyTicket = InComponentVerifyTicket;
InComponentVerifyTicket.INFO_TYPE = 'component_verify_ticket';
