// @tnwx/commons/dist/entity/msg/in/InExternalContact.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const BaseMsg_1 = require("../BaseMsg");
/**
 * @author Javen
 * @copyright javendev@126.com
 * @description 外部联系人事件
 */
class InExternalContact extends BaseMsg_1.BaseMsg {
    constructor(suiteId, authCorpId, infoType, timeStamp, changeType, userId, externalUserId, welcomeCode) {
        super();
        this.suiteid = suiteId;
        this.authcorpid = authCorpId;
        this.infotype = infoType;
        this.timestamp = timeStamp;
        this.changetype = changeType;
        this.userid = userId;
        this.externaluserid = externalUserId;
        this.welcomecode = welcomeCode;
    }
    get suiteId() {
        return this.suiteid;
    }
    set suiteId(suiteId) {
        this.suiteid = suiteId;
    }
    get authCorpId() {
        return this.authcorpid;
    }
    set authCorpId(authCorpId) {
        this.authcorpid = authCorpId;
    }
    get infoType() {
        return this.infotype;
    }
    set infoType(infoType) {
        this.infotype = infoType;
    }
    get timeStamp() {
        return this.timestamp;
    }
    set timeStamp(timeStamp) {
        this.timestamp = timeStamp;
    }
    get changeType() {
        return this.changetype;
    }
    set changeType(changeType) {
        this.changetype = changeType;
    }
    get userId() {
        return this.userid;
    }
    set userId(userId) {
        this.userid = userId;
    }
    get externalUserId() {
        return this.externaluserid;
    }
    set externalUserId(externalUserId) {
        this.externaluserid = externalUserId;
    }
    get welcomeCode() {
        return this.welcomecode;
    }
    set welcomeCode(welcomeCode) {
        this.welcomecode = welcomeCode;
    }
}
exports.InExternalContact = InExternalContact;
InExternalContact.INFO_TYPE = 'change_external_contact';
