// @tnwx/commons/dist/entity/msg/in/InLocationMsg.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
/**
 * @author Javen
 * @copyright javendev@126.com
 * @description  接收地理位置消息
 */
const InMsg_1 = require("./InMsg");
class InLocationMsg extends InMsg_1.InMsg {
    constructor(toUserName, fromUserName, createTime, msgType) {
        super(toUserName, fromUserName, createTime, msgType);
    }
    get getLocation_X() {
        return this.location_X;
    }
    set setLocation_X(location_X) {
        this.location_X = location_X;
    }
    get getLocation_Y() {
        return this.location_Y;
    }
    set setLocation_Y(location_Y) {
        this.location_Y = location_Y;
    }
    get getScale() {
        return this.scale;
    }
    set setScale(scale) {
        this.scale = scale;
    }
    get getLabel() {
        return this.label;
    }
    set setLabel(label) {
        this.label = label;
    }
    get getMsgId() {
        return this.msgId;
    }
    set setMsgId(msgId) {
        this.msgId = msgId;
    }
}
exports.InLocationMsg = InLocationMsg;
