// @tnwx/commons/dist/entity/msg/in/InMsg.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const BaseMsg_1 = require("../BaseMsg");
/**
 * @author Javen
 * @copyright javendev@126.com
 * @description 接收消息抽象类
 */
class InMsg extends BaseMsg_1.BaseMsg {
    constructor(toUserName, fromUserName, createTime, msgType) {
        super();
        this.toUserName = toUserName;
        this.fromUserName = fromUserName;
        this.createTime = createTime;
        this.msgType = msgType;
    }
    get getToUserName() {
        return this.toUserName;
    }
    set setToUserName(toUserName) {
        this.toUserName = toUserName;
    }
    get getFromUserName() {
        return this.fromUserName;
    }
    set setFromUserName(fromUserName) {
        this.fromUserName = fromUserName;
    }
    get getCreateTime() {
        return this.createTime;
    }
    set setCreateTime(createTime) {
        this.createTime = createTime;
    }
    get getMsgType() {
        return this.msgType;
    }
    set setMsgType(msgType) {
        this.msgType = msgType;
    }
    get getAgentId() {
        return this.agentId;
    }
    set setAgentId(agentId) {
        this.agentId = agentId;
    }
}
exports.InMsg = InMsg;
