// @tnwx/commons/dist/entity/msg/in/InRegisterCorp.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const BaseMsg_1 = require("../BaseMsg");
/**
 * @author Javen
 * @copyright javendev@126.com
 * @description 注册完成回调事件
 */
class InRegisterCorp extends BaseMsg_1.BaseMsg {
    constructor(serviceCorpId, infoType, timeStamp, registerCode, authCorpId, accessToken, expiresIn, userId, state, templateId) {
        super();
        this.servicecorpid = serviceCorpId;
        this.infotype = infoType;
        this.timestamp = timeStamp;
        this.registercode = registerCode;
        this.authcorpid = authCorpId;
        this.accesstoken = accessToken;
        this.expiresin = expiresIn;
        this.userid = userId;
        this.state = state;
        this.templateid = templateId;
    }
    get serviceCorpId() {
        return this.servicecorpid;
    }
    set serviceCorpId(serviceCorpId) {
        this.servicecorpid = serviceCorpId;
    }
    get infoType() {
        return this.infotype;
    }
    set infoType(infoType) {
        this.infotype = infoType;
    }
    get timeStamp() {
        return this.timestamp;
    }
    set timeStamp(timeStamp) {
        this.timestamp = timeStamp;
    }
    get registerCode() {
        return this.registercode;
    }
    set registerCode(registerCode) {
        this.registercode = registerCode;
    }
    get authCorpId() {
        return this.authcorpid;
    }
    set authCorpId(authCorpId) {
        this.authcorpid = authCorpId;
    }
    get accessToken() {
        return this.accesstoken;
    }
    set accessToken(accessToken) {
        this.accessToken = accessToken;
    }
    get expiresIn() {
        return this.expiresin;
    }
    set expiresIn(expiresIn) {
        this.expiresin = expiresIn;
    }
    get userId() {
        return this.userid;
    }
    set userId(userId) {
        this.userid = userId;
    }
    get getState() {
        return this.state;
    }
    set setState(state) {
        this.state = state;
    }
    get templateId() {
        return this.templateid;
    }
    set templateId(templateId) {
        this.templateid = templateId;
    }
}
exports.InRegisterCorp = InRegisterCorp;
InRegisterCorp.INFO_TYPE = 'register_corp';
