// @tnwx/commons/dist/entity/msg/in/InSuiteTicket.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const BaseMsg_1 = require("../BaseMsg");
/**
 * @author Javen
 * @copyright javendev@126.com
 * @description 推送suite_ticket
 */
class InSuiteTicket extends BaseMsg_1.BaseMsg {
    constructor(suiteId, infoType, timeStamp, suiteTicket) {
        super();
        this.suiteid = suiteId;
        this.infotype = infoType;
        this.timestamp = timeStamp;
        this.suiteticket = suiteTicket;
    }
    get suiteId() {
        return this.suiteid;
    }
    set suiteId(suiteId) {
        this.suiteid = suiteId;
    }
    get infoType() {
        return this.infotype;
    }
    set infoType(infoType) {
        this.infotype = infoType;
    }
    get timeStamp() {
        return this.timestamp;
    }
    set timeStamp(timeStamp) {
        this.timestamp = timeStamp;
    }
    get suiteTicket() {
        return this.suiteticket;
    }
    set suiteTicket(suiteTicket) {
        this.suiteticket = suiteTicket;
    }
}
exports.InSuiteTicket = InSuiteTicket;
InSuiteTicket.INFO_TYPE = 'suite_ticket';
