// @tnwx/commons/dist/entity/msg/in/event/InBatchJobResultEvent.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const EventInMsg_1 = require("./EventInMsg");
/**
 * @author Javen
 * @copyright javendev@126.com
 * @description 异步任务完成通知
 */
class InBatchJobResultEvent extends EventInMsg_1.EventInMsg {
    constructor(toUserName, fromUserName, createTime, event) {
        super(toUserName, fromUserName, createTime, event);
    }
    get getJobId() {
        return this.jobId;
    }
    set setJobId(jobId) {
        this.jobId = jobId;
    }
    set setJobType(jobType) {
        this.jobType = jobType;
    }
    get getJobType() {
        return this.jobType;
    }
    set setEventCode(eventCode) {
        this.eventCode = eventCode;
    }
    get getEventCode() {
        return this.eventCode;
    }
    set setEventMsg(eventMsg) {
        this.eventMsg = eventMsg;
    }
    get getEventMsg() {
        return this.eventMsg;
    }
}
exports.InBatchJobResultEvent = InBatchJobResultEvent;
InBatchJobResultEvent.EVENT = 'batch_job_result';
