// @tnwx/commons/dist/entity/msg/in/event/InExternalContactEvent.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const EventInMsg_1 = require("./EventInMsg");
/**
 * @author Javen
 * @copyright javendev@126.com
 * @description 企业客户事件
 */
class InExternalContactEvent extends EventInMsg_1.EventInMsg {
    constructor(toUserName, fromUserName, createTime, event) {
        super(toUserName, fromUserName, createTime, event);
    }
    get changeType() {
        return this.changetype;
    }
    set changeType(changeType) {
        this.changetype = changeType;
    }
    get userId() {
        return this.userid;
    }
    set userId(userId) {
        this.userid = userId;
    }
    get externalUserId() {
        return this.externaluserid;
    }
    set externalUserId(externalUserId) {
        this.externaluserid = externalUserId;
    }
    get getState() {
        return this.state;
    }
    set setState(state) {
        this.state = state;
    }
    get welcomeCode() {
        return this.welcomecode;
    }
    set welcomeCode(welcomeCode) {
        this.welcomecode = welcomeCode;
    }
}
exports.InExternalContactEvent = InExternalContactEvent;
InExternalContactEvent.EVENT = 'change_external_contact';
