// @tnwx/commons/dist/entity/msg/in/event/InMassEvent.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
/**
 * @author Javen
 * @copyright javendev@126.com
 * @description 群发结果
 */
const EventInMsg_1 = require("./EventInMsg");
class InMassEvent extends EventInMsg_1.EventInMsg {
    constructor(toUserName, fromUserName, createTime, event) {
        super(toUserName, fromUserName, createTime, event);
    }
    get getMsgId() {
        return this.msgId;
    }
    set setMsgId(msgId) {
        this.msgId = msgId;
    }
    get getStatus() {
        return this.status;
    }
    set setStatus(status) {
        this.status = status;
    }
    get getTotalCount() {
        return this.totalCount;
    }
    set setTotalCount(totalCount) {
        this.totalCount = totalCount;
    }
    get getFilterCount() {
        return this.filterCount;
    }
    set setFilterCount(filterCount) {
        this.filterCount = filterCount;
    }
    get getSentCount() {
        return this.sentCount;
    }
    set setSentCount(sentCount) {
        this.sentCount = sentCount;
    }
    get getErrorCount() {
        return this.errorCount;
    }
    set setErrorCount(errorCount) {
        this.errorCount = errorCount;
    }
}
exports.InMassEvent = InMassEvent;
InMassEvent.EVENT = 'MASSSENDJOBFINISH';
//群发成功
InMassEvent.EVENT_INMASS_STATUS_SENDSUCCESS = 'sendsuccess';
//群发失败，其他失败情况是err(num)
InMassEvent.EVENT_INMASS_STATUS_SENDFAIL = 'sendfail';
