// @tnwx/commons/dist/entity/msg/in/event/InMenuEvent.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const EventInMsg_1 = require("./EventInMsg");
class InMenuEvent extends EventInMsg_1.EventInMsg {
    constructor(toUserName, fromUserName, createTime, event) {
        super(toUserName, fromUserName, createTime, event);
    }
    get getEventKey() {
        return this.eventKey;
    }
    set setEventKey(eventKey) {
        this.eventKey = eventKey;
    }
    get getScanCodeInfo() {
        return this.scanCodeInfo;
    }
    set setScanCodeInfo(scanCodeInfo) {
        this.scanCodeInfo = scanCodeInfo;
    }
    get getMenuId() {
        return this.menuId;
    }
    set setMenuId(menuId) {
        this.menuId = menuId;
    }
}
exports.InMenuEvent = InMenuEvent;
// 1. 点击菜单拉取消息时的事件推送： CLICK
InMenuEvent.EVENT_INMENU_CLICK = 'CLICK';
// 2. 点击菜单跳转链接时的事件推送： VIEW
InMenuEvent.EVENT_INMENU_VIEW = 'VIEW';
// 3. scancode_push：扫码推事件
InMenuEvent.EVENT_INMENU_SCANCODE_PUSH = 'scancode_push';
// 4. scancode_waitmsg：扫码推事件且弹出“消息接收中”提示框
InMenuEvent.EVENT_INMENU_scancode_waitmsg = 'scancode_waitmsg';
// 5. pic_sysphoto：弹出系统拍照发图
InMenuEvent.EVENT_INMENU_PIC_SYSPHOTO = 'pic_sysphoto';
// 6. pic_photo_or_album：弹出拍照或者相册发图，先推送菜单事件，再推送图片消息
InMenuEvent.EVENT_INMENU_PIC_PHOTO_OR_ALBUM = 'pic_photo_or_album';
// 7. pic_weixin：弹出微信相册发图器
InMenuEvent.EVENT_INMENU_PIC_WEIXIN = 'pic_weixin';
// 8. location_select：弹出地理位置选择器
InMenuEvent.EVENT_INMENU_LOCATION_SELECT = 'location_select';
// 9. media_id：下发消息（除文本消息）
InMenuEvent.EVENT_INMENU_MEDIA_ID = 'media_id';
// 10. view_limited：跳转图文消息URL
InMenuEvent.EVENT_INMENU_VIEW_LIMITED = 'view_limited';
// 11. view_miniprogram：点击菜单跳转小程序
InMenuEvent.EVENT_INMENU_VIEW_MINIPROGRAM = 'view_miniprogram';
