// @tnwx/commons/dist/entity/msg/in/event/InShakearoundUserShakeEvent.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const InMsg_1 = require("../InMsg");
class InShakearoundUserShakeEvent extends InMsg_1.InMsg {
    constructor(toUserName, fromUserName, createTime, msgType) {
        super(toUserName, fromUserName, createTime, msgType);
    }
    get getEvent() {
        return this.event;
    }
    set setEvent(event) {
        this.event = event;
    }
    get getUuid() {
        return this.uuid;
    }
    set setUuid(uuid) {
        this.uuid = uuid;
    }
    get getMajor() {
        return this.major;
    }
    set setMajor(major) {
        this.major = major;
    }
    get getMinor() {
        return this.minor;
    }
    set setMinor(minor) {
        this.minor = minor;
    }
    get getDistance() {
        return this.distance;
    }
    set setDistance(distance) {
        this.distance = distance;
    }
    get getAroundBeaconList() {
        return this.aroundBeaconList;
    }
    set setAroundBeaconList(aroundBeaconList) {
        this.aroundBeaconList = aroundBeaconList;
    }
}
exports.InShakearoundUserShakeEvent = InShakearoundUserShakeEvent;
class AroundBeacon {
    constructor(uuid, major, minor, distance) {
        this.uuid = uuid;
        this.major = major;
        this.minor = minor;
        this.distance = distance;
    }
}
exports.AroundBeacon = AroundBeacon;
