// @tnwx/commons/dist/entity/msg/in/event/InUpdateUserEvent.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const EventInMsg_1 = require("./EventInMsg");
/**
 * @author Javen
 * @copyright javendev@126.com
 * @description 成员变更通知
 */
class InUpdateUserEvent extends EventInMsg_1.EventInMsg {
    constructor(toUserName, fromUserName, createTime, event) {
        super(toUserName, fromUserName, createTime, event);
    }
    get getChangeType() {
        return this.changeType;
    }
    set setChangeType(changeType) {
        this.changeType = changeType;
    }
    set setUserId(userId) {
        this.userId = userId;
    }
    get getUserId() {
        return this.userId;
    }
    set setNewUserId(newUserId) {
        this.newUserId = newUserId;
    }
    get getNewUserId() {
        return this.newUserId;
    }
    set setName(name) {
        this.name = name;
    }
    get getName() {
        return this.name;
    }
    set setDepartment(department) {
        this.department = department;
    }
    get getDepartment() {
        return this.department;
    }
    set setIsLeaderInDept(isLeaderInDept) {
        this.isLeaderInDept = isLeaderInDept;
    }
    get getIsLeaderInDept() {
        return this.isLeaderInDept;
    }
    set setPosition(position) {
        this.position = position;
    }
    get getPosition() {
        return this.position;
    }
    set setMobile(mobile) {
        this.mobile = mobile;
    }
    get getMobile() {
        return this.mobile;
    }
    set setGender(gender) {
        this.gender = gender;
    }
    get getGender() {
        return this.gender;
    }
    set setEmail(email) {
        this.email = email;
    }
    get getEmail() {
        return this.email;
    }
    set setStatus(status) {
        this.status = status;
    }
    get getStatus() {
        return this.status;
    }
    set setAvatar(avatar) {
        this.avatar = avatar;
    }
    get getAvatar() {
        return this.avatar;
    }
    set setAlias(alias) {
        this.alias = alias;
    }
    get getAlias() {
        return this.alias;
    }
    set setTelephone(telephone) {
        this.telephone = telephone;
    }
    get getTelephone() {
        return this.telephone;
    }
    set setAddress(address) {
        this.address = address;
    }
    get getAddress() {
        return this.address;
    }
    set setExtAttr(extAttr) {
        this.extAttr = extAttr;
    }
    get getExtAttr() {
        return this.extAttr;
    }
}
exports.InUpdateUserEvent = InUpdateUserEvent;
InUpdateUserEvent.EVENT = 'change_contact';
InUpdateUserEvent.CREATE_USER = 'create_user';
InUpdateUserEvent.UPDATE_USER = 'update_user';
InUpdateUserEvent.DELETE_USER = 'delete_user';
