// @tnwx/commons/dist/entity/msg/in/event/InUpdatePartyEvent.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const EventInMsg_1 = require("./EventInMsg");
/**
 * @author Javen
 * @copyright javendev@126.com
 * @description 部门变更通知
 */
class InUpdatePartyEvent extends EventInMsg_1.EventInMsg {
    constructor(toUserName, fromUserName, createTime, event) {
        super(toUserName, fromUserName, createTime, event);
    }
    get getChangeType() {
        return this.changeType;
    }
    set setChangeType(changeType) {
        this.changeType = changeType;
    }
    get getId() {
        return this.id;
    }
    set setId(id) {
        this.id = id;
    }
    set setName(name) {
        this.name = name;
    }
    get getName() {
        return this.name;
    }
    set setParentId(parentId) {
        this.parentId = parentId;
    }
    get getParentId() {
        return this.parentId;
    }
    set setOrder(order) {
        this.order = order;
    }
    get getOrder() {
        return this.order;
    }
}
exports.InUpdatePartyEvent = InUpdatePartyEvent;
InUpdatePartyEvent.CREATE_PARTY = 'create_party';
InUpdatePartyEvent.UPDATE_PARTY = 'update_party';
InUpdatePartyEvent.DELETE_PARTY = 'delete_party';
