// @tnwx/commons/dist/entity/msg/in/event/InUpdateTagEvent.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const EventInMsg_1 = require("./EventInMsg");
/**
 * @author Javen
 * @copyright javendev@126.com
 * @description 标签变更通知
 */
class InUpdateTagEvent extends EventInMsg_1.EventInMsg {
    constructor(toUserName, fromUserName, createTime, event) {
        super(toUserName, fromUserName, createTime, event);
    }
    get getTagId() {
        return this.tagId;
    }
    set setTagId(tagId) {
        this.tagId = tagId;
    }
    set setAddUserItems(addUserItems) {
        this.addUserItems = addUserItems;
    }
    get getAddUserItems() {
        return this.addUserItems;
    }
    set setDelUserItems(delUserItems) {
        this.delUserItems = delUserItems;
    }
    get getDelUserItems() {
        return this.delUserItems;
    }
    set setAddPartyItems(addPartyItems) {
        this.addPartyItems = addPartyItems;
    }
    get getAddPartyItems() {
        return this.addPartyItems;
    }
    set setDelPartyItems(delPartyItems) {
        this.delPartyItems = delPartyItems;
    }
    get getDelPartyItems() {
        return this.delPartyItems;
    }
}
exports.InUpdateTagEvent = InUpdateTagEvent;
InUpdateTagEvent.CHANGE_TYPE = 'update_tag';
