// @tnwx/commons/dist/entity/msg/in/event/InWifiEvent.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
/**
 * @author Javen
 * @copyright javendev@126.com
 * @description wifi连网后下发消息
 */
const EventInMsg_1 = require("./EventInMsg");
class InWifiEvent extends EventInMsg_1.EventInMsg {
    get getConnectTime() {
        return this.connectTime;
    }
    set setConnectTime(connectTime) {
        this.connectTime = connectTime;
    }
    get getExpireTime() {
        return this.expireTime;
    }
    set setExpireTime(expireTime) {
        this.expireTime = expireTime;
    }
    get getVendorId() {
        return this.vendorId;
    }
    set setVendorId(vendorId) {
        this.vendorId = vendorId;
    }
    get getShopId() {
        return this.shopId;
    }
    set setShopId(shopId) {
        this.shopId = shopId;
    }
    get getDeviceNo() {
        return this.deviceNo;
    }
    set setDeviceNo(deviceNo) {
        this.deviceNo = deviceNo;
    }
}
exports.InWifiEvent = InWifiEvent;
InWifiEvent.EVENT = 'WifiConnected';
