// @tnwx/commons/dist/entity/msg/in/card/InCardPayOrderEvent.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
/**
 * @author Javen
 * @copyright javendev@126.com
 * @description 券点流水详情事件
 */
const EventInMsg_1 = require("../event/EventInMsg");
class InCardPayOrderEvent extends EventInMsg_1.EventInMsg {
    constructor(toUserName, fromUserName, createTime, msgType) {
        super(toUserName, fromUserName, createTime, msgType);
    }
    get getOrderId() {
        return this.orderId;
    }
    set setOrderId(orderId) {
        this.orderId = orderId;
    }
    get getStatus() {
        return this.status;
    }
    set setStatus(status) {
        this.status = status;
    }
    get getCreateOrderTime() {
        return this.createOrderTime;
    }
    set setCreateOrderTime(createOrderTime) {
        this.createOrderTime = createOrderTime;
    }
    get getPayFinishTime() {
        return this.payFinishTime;
    }
    set setPayFinishTime(payFinishTime) {
        this.payFinishTime = payFinishTime;
    }
    get getDesc() {
        return this.desc;
    }
    set setDesc(desc) {
        this.desc = desc;
    }
    get getFreeCoinCount() {
        return this.freeCoinCount;
    }
    set setFreeCoinCount(freeCoinCount) {
        this.freeCoinCount = freeCoinCount;
    }
    get getPayCoinCount() {
        return this.payCoinCount;
    }
    set setPayCoinCount(payCoinCount) {
        this.payCoinCount = payCoinCount;
    }
    get getRefundFreeCoinCount() {
        return this.refundFreeCoinCount;
    }
    set setRefundFreeCoinCount(refundFreeCoinCount) {
        this.refundFreeCoinCount = refundFreeCoinCount;
    }
    get getRefundPayCoinCount() {
        return this.refundPayCoinCount;
    }
    set setRefundPayCoinCount(refundPayCoinCount) {
        this.refundPayCoinCount = refundPayCoinCount;
    }
    get getOrderType() {
        return this.orderType;
    }
    set setOrderType(orderType) {
        this.orderType = orderType;
    }
    get getMemo() {
        return this.memo;
    }
    set setMemo(memo) {
        this.memo = memo;
    }
    get getReceiptInfo() {
        return this.receiptInfo;
    }
    set setReceiptInfo(receiptInfo) {
        this.receiptInfo = receiptInfo;
    }
}
exports.InCardPayOrderEvent = InCardPayOrderEvent;
InCardPayOrderEvent.EVENT = 'card_pay_order';
