// @tnwx/commons/dist/entity/msg/in/card/InMerChantOrderEvent.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
/**
 * @author Javen
 * @copyright javendev@126.com
 * @description 微信小店订单支付成功接口消息
 */
const EventInMsg_1 = require("../event/EventInMsg");
class InMerChantOrderEvent extends EventInMsg_1.EventInMsg {
    constructor(toUserName, fromUserName, createTime, msgType) {
        super(toUserName, fromUserName, createTime, msgType);
    }
    get getOrderId() {
        return this.orderId;
    }
    set setOrderId(orderId) {
        this.orderId = orderId;
    }
    get getOrderStatus() {
        return this.orderStatus;
    }
    set setOrderStatus(orderStatus) {
        this.orderStatus = orderStatus;
    }
    get getProductId() {
        return this.productId;
    }
    set setProductId(productId) {
        this.productId = productId;
    }
    get getSkuInfo() {
        return this.skuInfo;
    }
    set setSkuInfo(skuInfo) {
        this.skuInfo = skuInfo;
    }
}
exports.InMerChantOrderEvent = InMerChantOrderEvent;
InMerChantOrderEvent.EVENT = 'merchant_order';
