// @tnwx/commons/dist/entity/msg/in/card/InUpdateMemberCardEvent.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
/**
 * @author Javen
 * @copyright javendev@126.com
 * @description 微信会员卡积分变更
 */
const EventInMsg_1 = require("../event/EventInMsg");
class InUpdateMemberCardEvent extends EventInMsg_1.EventInMsg {
    constructor(toUserName, fromUserName, createTime, msgType) {
        super(toUserName, fromUserName, createTime, msgType);
    }
    get getCardId() {
        return this.cardId;
    }
    set setCardId(cardId) {
        this.cardId = cardId;
    }
    get getUserCardCode() {
        return this.userCardCode;
    }
    set setUserCardCode(userCardCode) {
        this.userCardCode = userCardCode;
    }
    get getModifyBalance() {
        return this.modifyBalance;
    }
    set setModifyBalance(modifyBalance) {
        this.modifyBalance = modifyBalance;
    }
    get getModifyBonus() {
        return this.modifyBonus;
    }
    set setModifyBonus(modifyBonus) {
        this.modifyBonus = modifyBonus;
    }
}
exports.InUpdateMemberCardEvent = InUpdateMemberCardEvent;
InUpdateMemberCardEvent.EVENT = 'update_member_card';
