// @tnwx/commons/dist/entity/msg/in/card/InUserCardEvent.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
/**
 * @author Javen
 * @copyright javendev@126.com
 * @description 微信会员卡卡券
 */
const EventInMsg_1 = require("../event/EventInMsg");
class InUserCardEvent extends EventInMsg_1.EventInMsg {
    constructor(toUserName, fromUserName, createTime, msgType) {
        super(toUserName, fromUserName, createTime, msgType);
    }
    get getCardId() {
        return this.cardId;
    }
    set setCardId(cardId) {
        this.cardId = cardId;
    }
    get getUserCardCode() {
        return this.userCardCode;
    }
    set setUserCardCode(userCardCode) {
        this.userCardCode = userCardCode;
    }
}
exports.InUserCardEvent = InUserCardEvent;
// 微信会员卡激活接口
InUserCardEvent.EVENT_MEMBERCARD = 'submit_membercard_user_info';
// 微信会员卡二维码扫描领取接口
InUserCardEvent.EVENT_USER_VIEW = 'user_view_card';
// 从卡券进入公众号会话事件推送
InUserCardEvent.EVENT_USER_ENTER = 'user_enter_session_from_card';
// 卡券删除事件推送
InUserCardEvent.EVENT_USER_DEL = 'user_del_card';
