// @tnwx/commons/dist/entity/msg/in/card/InUserPayFromCardEvent.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
/**
 * @author Javen
 * @copyright javendev@126.com
 * @description 微信会员卡快速买单
 */
const EventInMsg_1 = require("../event/EventInMsg");
class InUserPayFromCardEvent extends EventInMsg_1.EventInMsg {
    constructor(toUserName, fromUserName, createTime, msgType) {
        super(toUserName, fromUserName, createTime, msgType);
    }
    get getCardId() {
        return this.cardId;
    }
    set setCardId(cardId) {
        this.cardId = cardId;
    }
    get getUserCardCode() {
        return this.userCardCode;
    }
    set setUserCardCode(userCardCode) {
        this.userCardCode = userCardCode;
    }
    get getTransId() {
        return this.transId;
    }
    set setTransId(transId) {
        this.transId = transId;
    }
    get getLocationId() {
        return this.locationId;
    }
    set setLocationId(locationId) {
        this.locationId = locationId;
    }
    get getFee() {
        return this.fee;
    }
    set setFee(fee) {
        this.fee = fee;
    }
    get getOriginalFee() {
        return this.originalFee;
    }
    set setOriginalFee(originalFee) {
        this.originalFee = originalFee;
    }
}
exports.InUserPayFromCardEvent = InUserPayFromCardEvent;
InUserPayFromCardEvent.EVENT = 'user_pay_from_pay_cell';
