// @tnwx/commons/dist/entity/msg/in/card/InUserGetCardEvent.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
/**
 * @author Javen
 * @copyright javendev@126.com
 * @description 领取事件推送
 */
const EventInMsg_1 = require("../event/EventInMsg");
class InUserGetCardEvent extends EventInMsg_1.EventInMsg {
    constructor(toUserName, fromUserName, createTime, msgType) {
        super(toUserName, fromUserName, createTime, msgType);
    }
    get getCardId() {
        return this.cardId;
    }
    set setCardId(cardId) {
        this.cardId = cardId;
    }
    get getIsGiveByFriend() {
        return this.isGiveByFriend;
    }
    set setIsGiveByFriend(isGiveByFriend) {
        this.isGiveByFriend = isGiveByFriend;
    }
    get getUserCardCode() {
        return this.userCardCode;
    }
    set setUserCardCode(userCardCode) {
        this.userCardCode = userCardCode;
    }
    get getOuterId() {
        return this.outerId;
    }
    set setOuterId(outerId) {
        this.outerId = outerId;
    }
    get getFriendUserName() {
        return this.friendUserName;
    }
    set setFriendUserName(friendUserName) {
        this.friendUserName = friendUserName;
    }
    get getOldUserCardCode() {
        return this.oldUserCardCode;
    }
    set setOldUserCardCode(oldUserCardCode) {
        this.oldUserCardCode = oldUserCardCode;
    }
    get getOuterStr() {
        return this.outerStr;
    }
    set setOuterStr(outerStr) {
        this.outerStr = outerStr;
    }
    get getIsRestoreMemberCard() {
        return this.isRestoreMemberCard;
    }
    set setIsRestoreMemberCard(isRestoreMemberCard) {
        this.isRestoreMemberCard = isRestoreMemberCard;
    }
    get getIsRecommendByFriend() {
        return this.isRecommendByFriend;
    }
    set setIsRecommendByFriend(isRecommendByFriend) {
        this.isRecommendByFriend = isRecommendByFriend;
    }
}
exports.InUserGetCardEvent = InUserGetCardEvent;
InUserGetCardEvent.EVENT = 'user_get_card';
