// @tnwx/commons/dist/entity/msg/in/card/InUserGiftingCardEvent.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
/**
 * @author Javen
 * @copyright javendev@126.com
 * @description 转赠事件推送
 */
const EventInMsg_1 = require("../event/EventInMsg");
class InUserGiftingCardEvent extends EventInMsg_1.EventInMsg {
    constructor(toUserName, fromUserName, createTime, msgType) {
        super(toUserName, fromUserName, createTime, msgType);
    }
    get getCardId() {
        return this.cardId;
    }
    set setCardId(cardId) {
        this.cardId = cardId;
    }
    get getUserCardCode() {
        return this.userCardCode;
    }
    set setUserCardCode(userCardCode) {
        this.userCardCode = userCardCode;
    }
    get getIsReturnBack() {
        return this.isReturnBack;
    }
    set setIsReturnBack(isReturnBack) {
        this.isReturnBack = isReturnBack;
    }
    get getFriendUserName() {
        return this.friendUserName;
    }
    set setFriendUserName(friendUserName) {
        this.friendUserName = friendUserName;
    }
    get getIsChatRoom() {
        return this.isChatRoom;
    }
    set setIsChatRoom(isChatRoom) {
        this.isChatRoom = isChatRoom;
    }
}
exports.InUserGiftingCardEvent = InUserGiftingCardEvent;
InUserGiftingCardEvent.EVENT = 'user_gifting_card';
