// @tnwx/commons/dist/entity/msg/in/card/InUserConsumeCardEvent.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
/**
 * @author Javen
 * @copyright javendev@126.com
 * @description 核销事件推送
 */
const EventInMsg_1 = require("../event/EventInMsg");
class InUserConsumeCardEvent extends EventInMsg_1.EventInMsg {
    constructor(toUserName, fromUserName, createTime, msgType) {
        super(toUserName, fromUserName, createTime, msgType);
    }
    get getCardId() {
        return this.cardId;
    }
    set setCardId(cardId) {
        this.cardId = cardId;
    }
    get getUserCardCode() {
        return this.userCardCode;
    }
    set setUserCardCode(userCardCode) {
        this.userCardCode = userCardCode;
    }
    get getConsumeSource() {
        return this.consumeSource;
    }
    set setConsumeSource(consumeSource) {
        this.consumeSource = consumeSource;
    }
    get getLocationName() {
        return this.locationName;
    }
    set setLocationName(locationName) {
        this.locationName = locationName;
    }
    get getStaffOpenId() {
        return this.staffOpenId;
    }
    set setStaffOpenId(staffOpenId) {
        this.staffOpenId = staffOpenId;
    }
    get getVerifyCode() {
        return this.verifyCode;
    }
    set setVerifyCode(verifyCode) {
        this.verifyCode = verifyCode;
    }
    get getRemarkAmount() {
        return this.remarkAmount;
    }
    set setRemarkAmount(remarkAmount) {
        this.remarkAmount = remarkAmount;
    }
    get getOuterStr() {
        return this.outerStr;
    }
    set setOuterStr(outerStr) {
        this.outerStr = outerStr;
    }
}
exports.InUserConsumeCardEvent = InUserConsumeCardEvent;
InUserConsumeCardEvent.EVENT = 'user_consume_card';
