// @tnwx/miniprogram/dist/ImgApi.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
const util = require("util");
const accesstoken_1 = require("@tnwx/accesstoken");
const kits_1 = require("@tnwx/kits");
/**
 * @author Javen
 * @copyright javendev@126.com
 * @description 图像处理相关 API
 */
class ImgApi {
    /**
     * 图像处理
     * @param type 接口URL
     * @param imgUrl 图片的URL
     */
    static imgProcessingByUrl(type, imgUrl) {
        return __awaiter(this, void 0, void 0, function* () {
            let accessToken = yield accesstoken_1.AccessTokenApi.getAccessToken();
            let url = util.format(type, accessToken.getAccessToken);
            return kits_1.HttpKit.getHttpDelegate.httpGet(url.concat('&img_url=').concat(imgUrl));
        });
    }
    /**
     * 图像处理
     * @param type 接口URL
     * @param filePath 图片文件路径
     */
    static imgProcessingByFile(type, filePath) {
        return __awaiter(this, void 0, void 0, function* () {
            let accessToken = yield accesstoken_1.AccessTokenApi.getAccessToken();
            let url = util.format(type, accessToken.getAccessToken);
            return kits_1.HttpKit.getHttpDelegate.upload(url, filePath);
        });
    }
}
exports.ImgApi = ImgApi;
var ImgProcessingType;
(function (ImgProcessingType) {
    // 图片智能裁剪
    ImgProcessingType["AICROP"] = "https://api.weixin.qq.com/cv/img/aicrop?access_token=%s";
    // 条码/二维码识别
    ImgProcessingType["QRCODE"] = "https://api.weixin.qq.com/cv/img/qrcode?access_token=%s";
    // 图片高清化
    ImgProcessingType["SUPERRESOLUTION"] = "https://api.weixin.qq.com/cv/img/superresolution?access_token=%s";
})(ImgProcessingType = exports.ImgProcessingType || (exports.ImgProcessingType = {}));
