// @tnwx/miniprogram/dist/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var MiniProgramApi_1 = require("./MiniProgramApi");
exports.MiniProgramMediaType = MiniProgramApi_1.MiniProgramMediaType;
exports.MiniProgramApi = MiniProgramApi_1.MiniProgramApi;
var MiniSubscribeMsgApi_1 = require("./MiniSubscribeMsgApi");
exports.MiniSubscribeMsgApi = MiniSubscribeMsgApi_1.MiniSubscribeMsgApi;
var OCRApi_1 = require("./OCRApi");
exports.OCRApi = OCRApi_1.OCRApi;
exports.OCRType = OCRApi_1.OCRType;
var ImgApi_1 = require("./ImgApi");
exports.ImgApi = ImgApi_1.ImgApi;
exports.ImgProcessingType = ImgApi_1.ImgProcessingType;
var MiniCustomerMsgApi_1 = require("./MiniCustomerMsgApi");
exports.MiniCustomerMsgApi = MiniCustomerMsgApi_1.MiniCustomerMsgApi;
exports.MiniCSRMsgType = MiniCustomerMsgApi_1.MiniCSRMsgType;
var NearByPoiApi_1 = require("./NearByPoiApi");
exports.NearByPoiApi = NearByPoiApi_1.NearByPoiApi;
var MiniPluginManagerApi_1 = require("./MiniPluginManagerApi");
exports.MiniPluginManagerApi = MiniPluginManagerApi_1.MiniPluginManagerApi;
