// @tnwx/miniprogram/dist/OCRApi.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
const util = require("util");
const accesstoken_1 = require("@tnwx/accesstoken");
const kits_1 = require("@tnwx/kits");
/**
 * @author Javen
 * @copyright javendev@126.com
 * @description OCR 识别相关 API
 */
class OCRApi {
    /**
     * OCR 识别
     * @param OcrType 接口URL
     * @param imgUrl 图片的URL
     */
    static ocrByUrl(OcrType, imgUrl) {
        return __awaiter(this, void 0, void 0, function* () {
            let accessToken = yield accesstoken_1.AccessTokenApi.getAccessToken();
            let url = util.format(OcrType, accessToken.getAccessToken);
            return kits_1.HttpKit.getHttpDelegate.httpGet(url.concat('&img_url=').concat(imgUrl));
        });
    }
    /**
     * OCR 识别
     * @param ocrType 接口URL
     * @param filePath 图片的文件路径
     */
    static ocrByFile(ocrType, filePath) {
        return __awaiter(this, void 0, void 0, function* () {
            let accessToken = yield accesstoken_1.AccessTokenApi.getAccessToken();
            let url = util.format(ocrType, accessToken.getAccessToken);
            return kits_1.HttpKit.getHttpDelegate.upload(url, filePath);
        });
    }
}
exports.OCRApi = OCRApi;
var OCRType;
(function (OCRType) {
    // 银行卡 OCR 识别
    OCRType["BANKCARD"] = "https://api.weixin.qq.com/cv/ocr/bankcard?access_token=%s";
    // 营业执照 OCR 识别
    OCRType["BUSINESSLICENSE"] = "https://api.weixin.qq.com/cv/ocr/bizlicense?access_token=%s";
    // 驾驶证 OCR 识别
    OCRType["DRIVERLICENSE"] = "https://api.weixin.qq.com/cv/ocr/drivinglicense?access_token=%s";
    // 身份证 OCR 识别
    OCRType["IDCARD"] = "https://api.weixin.qq.com/cv/ocr/idcard?access_token=%s";
    // 通用印刷体 OCR 识别
    OCRType["PRINTEDTEXT"] = "https://api.weixin.qq.com/cv/ocr/comm?access_token=%s";
    // 行驶证 OCR 识别
    OCRType["VEHICLELICENSE"] = "https://api.weixin.qq.com/cv/ocr/driving?access_token=%s";
})(OCRType = exports.OCRType || (exports.OCRType = {}));
