// @tnwx/commons/dist/entity/subscribe/SubscribeMsg.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
/**
 * @author Javen
 * @copyright javendev@126.com
 * @description 订阅消息
 */
class SubscribeMsg {
    constructor(touser, template_id, scene, title, data) {
        this.touser = touser;
        this.template_id = template_id;
        this.scene = scene;
        this.title = title;
        this.data = data;
    }
    get getUrl() {
        return this.url;
    }
    set setUrl(url) {
        this.url = url;
    }
    get getMiniProgram() {
        return this.miniprogram;
    }
    set setMiniProgram(miniprogram) {
        this.miniprogram = miniprogram;
    }
}
exports.SubscribeMsg = SubscribeMsg;
class MiniProgram {
    constructor(appid, pagepath) {
        this.appid = appid;
        this.pagepath = pagepath;
    }
}
exports.MiniProgram = MiniProgram;
class Data {
    constructor(content) {
        this.content = content;
    }
}
exports.Data = Data;
class Content {
    constructor(value, color) {
        this.value = value;
        this.color = color;
    }
}
exports.Content = Content;
