// @tnwx/commons/dist/entity/JsTicket.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
class JsTicket {
    constructor(json) {
        this.json = json;
        let ticket = JSON.parse(json);
        this.ticket = ticket.ticket;
        this.expires_in = ticket.expires_in;
        this.errcode = ticket.errcode;
        this.errmsg = ticket.errmsg;
        if (this.expires_in) {
            this.expired_time = new Date().getTime() + (this.expires_in - 9) * 1000;
        }
        // 从缓存读取时还原
        if (ticket.expired_time) {
            this.expired_time = ticket.expired_time;
        }
    }
    get getCacheJson() {
        let temp = JSON.parse(this.json);
        temp.expired_time = this.expired_time;
        return JSON.stringify(temp);
    }
    isAvailable() {
        if (this.expired_time == null)
            return false;
        if (this.errcode != null)
            return false;
        if (this.expired_time < new Date().getTime())
            return false;
        return this.ticket != null;
    }
    get getTicket() {
        return this.ticket;
    }
    set setTicket(ticket) {
        this.ticket = ticket;
    }
    get getExpiresIn() {
        return this.expires_in;
    }
    set setExpiresIn(expires_in) {
        this.expires_in = expires_in;
    }
    get getErrCode() {
        return this.errcode;
    }
    set setErrCode(errcode) {
        this.errcode = errcode;
    }
    get getErrMsg() {
        return this.errmsg;
    }
    set setErrMsg(errmsg) {
        this.errmsg = errmsg;
    }
    get getJson() {
        return this.json;
    }
}
exports.JsTicket = JsTicket;
