// @tnwx/commons/dist/entity/msg/out/OutMsg.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
class OutMsg {
    constructor(inMsg) {
        this.toUserName = inMsg.getFromUserName;
        this.fromUserName = inMsg.getToUserName;
        this.createTime = this.now();
    }
    toXml() {
        let xmlContent = '<xml>\n<ToUserName><![CDATA[' + this.toUserName + ']]></ToUserName>\n';
        xmlContent += '<FromUserName><![CDATA[' + this.fromUserName + ']]></FromUserName>\n';
        xmlContent += '<CreateTime>' + this.createTime + '</CreateTime>\n';
        xmlContent += '<MsgType><![CDATA[' + this.msgType + ']]></MsgType>\n';
        return xmlContent;
    }
    now() {
        return new Date().getTime();
    }
    get getToUserName() {
        return this.toUserName;
    }
    set setToUserName(toUserName) {
        this.toUserName = toUserName;
    }
    get getFromUserName() {
        return this.fromUserName;
    }
    set setFromUserName(fromUserName) {
        this.fromUserName = fromUserName;
    }
    get getCreateTime() {
        return this.createTime;
    }
    set setCreateTime(createTime) {
        this.createTime = createTime;
    }
    get getMsgType() {
        return this.msgType;
    }
    set setMsgType(msgType) {
        this.msgType = msgType;
    }
}
exports.OutMsg = OutMsg;
