// @tnwx/wxmp/dist/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var WeChat_1 = require("./WeChat");
exports.WeChat = WeChat_1.WeChat;
var AutoReplyInfoApi_1 = require("./api/AutoReplyInfoApi");
exports.AutoReplyInfoApi = AutoReplyInfoApi_1.AutoReplyInfoApi;
var CallbackApi_1 = require("./api/CallbackApi");
exports.CallbackApi = CallbackApi_1.CallbackApi;
exports.CheckAction = CallbackApi_1.CheckAction;
exports.CheckOperator = CallbackApi_1.CheckOperator;
var CardApi_1 = require("./api/CardApi");
exports.CardApi = CardApi_1.CardApi;
var CustomServiceApi_1 = require("./api/CustomServiceApi");
exports.CustomServiceApi = CustomServiceApi_1.CustomServiceApi;
var DatacubeApi_1 = require("./api/DatacubeApi");
exports.DatacubeApi = DatacubeApi_1.DatacubeApi;
var MediaApi_1 = require("./api/MediaApi");
exports.MediaApi = MediaApi_1.MediaApi;
exports.MediaType = MediaApi_1.MediaType;
exports.MediaArticles = MediaApi_1.MediaArticles;
var MemberCardApi_1 = require("./api/MemberCardApi");
exports.MemberCardApi = MemberCardApi_1.MemberCardApi;
var MenuApi_1 = require("./api/MenuApi");
exports.MenuApi = MenuApi_1.MenuApi;
var MessageApi_1 = require("./api/MessageApi");
exports.MessageApi = MessageApi_1.MessageApi;
var PoiApi_1 = require("./api/PoiApi");
exports.PoiApi = PoiApi_1.PoiApi;
var QrcodeApi_1 = require("./api/QrcodeApi");
exports.QrcodeApi = QrcodeApi_1.QrcodeApi;
var SemanticApi_1 = require("./api/SemanticApi");
exports.SemanticApi = SemanticApi_1.SemanticApi;
var ShortUrlApi_1 = require("./api/ShortUrlApi");
exports.ShortUrlApi = ShortUrlApi_1.ShortUrlApi;
var SnsAccessTokenApi_1 = require("./api/SnsAccessTokenApi");
exports.SnsAccessTokenApi = SnsAccessTokenApi_1.SnsAccessTokenApi;
var SubscribeMsgApi_1 = require("./api/SubscribeMsgApi");
exports.SubscribeMsgApi = SubscribeMsgApi_1.SubscribeMsgApi;
var TagApi_1 = require("./api/TagApi");
exports.TagApi = TagApi_1.TagApi;
var TemplateApi_1 = require("./api/TemplateApi");
exports.TemplateApi = TemplateApi_1.TemplateApi;
var UserApi_1 = require("./api/UserApi");
exports.UserApi = UserApi_1.UserApi;
exports.BatchUserInfo = UserApi_1.BatchUserInfo;
var ShakeAroundApi_1 = require("./api/shakearound/ShakeAroundApi");
exports.ShakeAroundApi = ShakeAroundApi_1.ShakeAroundApi;
exports.MaterialType = ShakeAroundApi_1.MaterialType;
var ShakeAroundDeviceApi_1 = require("./api/shakearound/ShakeAroundDeviceApi");
exports.ShakeAroundDeviceApi = ShakeAroundDeviceApi_1.ShakeAroundDeviceApi;
exports.DeviceIdentifier = ShakeAroundDeviceApi_1.DeviceIdentifier;
var ShakeAroundPageApi_1 = require("./api/shakearound/ShakeAroundPageApi");
exports.ShakeAroundPageApi = ShakeAroundPageApi_1.ShakeAroundPageApi;
var ShakeAroundStatisticsApi_1 = require("./api/shakearound/ShakeAroundStatisticsApi");
exports.ShakeAroundStatisticsApi = ShakeAroundStatisticsApi_1.ShakeAroundStatisticsApi;
var ShakeAroundUserApi_1 = require("./api/shakearound/ShakeAroundUserApi");
exports.ShakeAroundUserApi = ShakeAroundUserApi_1.ShakeAroundUserApi;
