// @tnwx/wxcp/dist/entity/QyBaseMsg.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
/**
 * @author Javen
 * @copyright javendev@126.com
 * @description 主动发送消息抽象类
 */
class QyBaseMsg {
    constructor(msgType, agentId, toUser, toParty, toTag, safe = 0, enableIdTrans = 0, enableDuplicateCheck = 0, duplicateCheckInterval = 1800) {
        this.msgtype = msgType;
        this.agentid = agentId;
        this.touser = toUser;
        this.totag = toTag;
        this.toparty = toParty;
        this.safe = safe;
        this.enable_id_trans = enableIdTrans;
        this.enable_duplicate_check = enableDuplicateCheck;
        this.duplicate_check_interval = duplicateCheckInterval;
    }
    set chatId(chatid) {
        this.chatid = chatid;
    }
    get chatId() {
        return this.chatid;
    }
    set toUser(toUser) {
        this.touser = toUser;
    }
    get toUser() {
        return this.touser;
    }
    set toParty(toParty) {
        this.toparty = toParty;
    }
    get toParty() {
        return this.toparty;
    }
    set toTag(toTag) {
        this.totag = toTag;
    }
    get toTag() {
        return this.totag;
    }
    set msgType(msgType) {
        this.msgtype = msgType;
    }
    get msgType() {
        return this.msgtype;
    }
    set agentId(agentId) {
        this.agentid = agentId;
    }
    get agentId() {
        return this.agentid;
    }
    set setSafe(safe) {
        this.safe = safe;
    }
    get getSafe() {
        return this.safe;
    }
    set enableIdTrans(enableIdTrans) {
        this.enable_id_trans = enableIdTrans;
    }
    get enableIdTrans() {
        return this.enable_id_trans;
    }
    set enableDuplicateCheck(enableDuplicateCheck) {
        this.enable_duplicate_check = enableDuplicateCheck;
    }
    get enableDuplicateCheck() {
        return this.enable_duplicate_check;
    }
    set duplicateCheckInterval(duplicateCheckInterval) {
        this.duplicate_check_interval = duplicateCheckInterval;
    }
    get duplicateCheckInterval() {
        return this.duplicate_check_interval;
    }
}
exports.QyBaseMsg = QyBaseMsg;
class QyMediaId {
    constructor(mediaId) {
        this.media_id = mediaId;
    }
    set mediaId(mediaId) {
        this.media_id = mediaId;
    }
    get mediaId() {
        return this.media_id;
    }
}
exports.QyMediaId = QyMediaId;
