// @tnwx/wxcp/dist/entity/QyImageMsg.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const QyBaseMsg_1 = require("./QyBaseMsg");
const QySendMsgApi_1 = require("../QySendMsgApi");
/**
 * @author Javen
 * @copyright javendev@126.com
 * @description 图片消息
 */
class QyImageMsg extends QyBaseMsg_1.QyBaseMsg {
    constructor(image, agentId, toUser, toParty, toTag, safe = 0, enableIdTrans = 0, enableDuplicateCheck = 0, duplicateCheckInterval = 1800) {
        super(QySendMsgApi_1.QySendMessageType.IMAGE, agentId, toUser, toParty, toTag, safe, enableIdTrans, enableDuplicateCheck, duplicateCheckInterval);
        this.image = image;
    }
    get getImage() {
        return this.image;
    }
    set setImage(image) {
        this.image = image;
    }
}
exports.QyImageMsg = QyImageMsg;
class QyImage {
    constructor(base64, md5) {
        this.base64 = base64;
        this.md5 = md5;
    }
    set setBase64(base64) {
        this.base64 = base64;
    }
    get getBase64() {
        return this.base64;
    }
    set setMd5(md5) {
        this.md5 = md5;
    }
    get getMd5() {
        return this.md5;
    }
}
exports.QyImage = QyImage;
