// @tnwx/wxcp/dist/entity/QyMiniProgramNoticeMsg.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const QyBaseMsg_1 = require("./QyBaseMsg");
const QySendMsgApi_1 = require("../QySendMsgApi");
/**
 * @author Javen
 * @copyright javendev@126.com
 * @description 小程序通知消息
 */
class QyMiniProgramNoticeMsg extends QyBaseMsg_1.QyBaseMsg {
    constructor(miniprogramNotice, agentId, toUser, toParty, toTag, safe = 0, enableIdTrans = 0, enableDuplicateCheck = 0, duplicateCheckInterval = 1800) {
        super(QySendMsgApi_1.QySendMessageType.MINIPROGRAM_NOTICE, agentId, toUser, toParty, toTag, safe, enableIdTrans, enableDuplicateCheck, duplicateCheckInterval);
        this.miniprogram_notice = miniprogramNotice;
    }
    get miniprogramNotice() {
        return this.miniprogram_notice;
    }
    set miniprogramNotice(miniprogramNotice) {
        this.miniprogram_notice = miniprogramNotice;
    }
}
exports.QyMiniProgramNoticeMsg = QyMiniProgramNoticeMsg;
class QyMiniprogram {
    constructor(appId, title, contentTtem, page, description, emphasisFirstItem) {
        this.appid = appId;
        this.title = title;
        this.content_item = contentTtem;
        this.page = page;
        this.emphasis_first_item = emphasisFirstItem;
        this.description = description;
    }
    set appId(appId) {
        this.appid = appId;
    }
    get appId() {
        return this.appid;
    }
    set setPage(page) {
        this.page = page;
    }
    get getPage() {
        return this.page;
    }
    set setTitle(title) {
        this.title = title;
    }
    get getTitle() {
        return this.title;
    }
    set contentItem(contentItem) {
        this.content_item = contentItem;
    }
    get contentItem() {
        return this.content_item;
    }
    set setDescription(description) {
        this.description = description;
    }
    get getDescription() {
        return this.description;
    }
    set emphasisFirstItem(emphasisFirstItem) {
        this.emphasis_first_item = emphasisFirstItem;
    }
    get emphasisFirstItem() {
        return this.emphasis_first_item;
    }
}
exports.QyMiniprogram = QyMiniprogram;
class QyKv {
    constructor(key, value) {
        this.key = key;
        this.value = value;
    }
    set setKey(key) {
        this.key = key;
    }
    get getKey() {
        return this.key;
    }
    set setValue(v) {
        this.value = v;
    }
    get getValue() {
        return this.value;
    }
}
exports.QyKv = QyKv;
