// @tnwx/wxcp/dist/entity/QyMpNewsMsg.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const QyBaseMsg_1 = require("./QyBaseMsg");
const QySendMsgApi_1 = require("../QySendMsgApi");
/**
 * @author Javen
 * @copyright javendev@126.com
 * @description 图文消息 mpnews
 */
class QyMpNewsMsg extends QyBaseMsg_1.QyBaseMsg {
    constructor(mpnews, agentId, toUser, toParty, toTag, safe = 0, enableIdTrans = 0, enableDuplicateCheck = 0, duplicateCheckInterval = 1800) {
        super(QySendMsgApi_1.QySendMessageType.MPNEWS, agentId, toUser, toParty, toTag, safe, enableIdTrans, enableDuplicateCheck, duplicateCheckInterval);
        this.mpnews = mpnews;
    }
    get mpNews() {
        return this.mpnews;
    }
    set mpNews(mpnews) {
        this.mpnews = mpnews;
    }
}
exports.QyMpNewsMsg = QyMpNewsMsg;
class QyMpNewsArticles {
    constructor(articles) {
        this.articles = articles;
    }
    set setArticles(articles) {
        this.articles = articles;
    }
    get getArticles() {
        return this.articles;
    }
}
exports.QyMpNewsArticles = QyMpNewsArticles;
class QyMpNews {
    constructor(title, thumbMediaId, content, author, contentSourceUrl, digest) {
        this.title = title;
        this.thumb_media_id = thumbMediaId;
        this.author = author;
        this.content = content;
        this.digest = digest;
    }
    get getTitle() {
        return this.title;
    }
    set setTitle(title) {
        this.title = title;
    }
    get thumbMediaId() {
        return this.thumb_media_id;
    }
    set thumbMediaId(thumbMediaId) {
        this.thumb_media_id = thumbMediaId;
    }
    get getAuthor() {
        return this.author;
    }
    set setAuthor(author) {
        this.author = author;
    }
    get contentSourceUrl() {
        return this.content_source_url;
    }
    set contentSourceUrl(contentSourceUrl) {
        this.content_source_url = contentSourceUrl;
    }
    get getContent() {
        return this.content;
    }
    set setContent(content) {
        this.content = content;
    }
    get getDigest() {
        return this.digest;
    }
    set setDigest(digest) {
        this.digest = digest;
    }
}
exports.QyMpNews = QyMpNews;
