// @tnwx/wxcp/dist/entity/QyNewsMsg.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const QyBaseMsg_1 = require("./QyBaseMsg");
const QySendMsgApi_1 = require("../QySendMsgApi");
/**
 * @author Javen
 * @copyright javendev@126.com
 * @description 图文消息
 */
class QyNewsMsg extends QyBaseMsg_1.QyBaseMsg {
    constructor(news, agentId, toUser, toParty, toTag, safe = 0, enableIdTrans = 0, enableDuplicateCheck = 0, duplicateCheckInterval = 1800) {
        super(QySendMsgApi_1.QySendMessageType.NEWS, agentId, toUser, toParty, toTag, safe, enableIdTrans, enableDuplicateCheck, duplicateCheckInterval);
        this.news = news;
    }
    get getNews() {
        return this.news;
    }
    set setNews(news) {
        this.news = news;
    }
}
exports.QyNewsMsg = QyNewsMsg;
class QyArticles {
    constructor(articles) {
        this.articles = articles;
    }
    set setArticles(articles) {
        this.articles = articles;
    }
    get getArticles() {
        return this.articles;
    }
}
exports.QyArticles = QyArticles;
class QyNews {
    constructor(title, description, picUrl, url) {
        this.title = title;
        this.description = description;
        this.picurl = picUrl;
        this.url = url;
    }
    get getTitle() {
        return this.title;
    }
    set setTitle(title) {
        this.title = title;
    }
    get getDescription() {
        return this.description;
    }
    set setDescription(description) {
        this.description = description;
    }
    get picUrl() {
        return this.picurl;
    }
    set picUrl(picUrl) {
        this.picurl = picUrl;
    }
    get getUrl() {
        return this.url;
    }
    set setUrl(url) {
        this.url = url;
    }
}
exports.QyNews = QyNews;
