// @tnwx/wxcp/dist/entity/QyTaskCardMsg.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const QyBaseMsg_1 = require("./QyBaseMsg");
const QySendMsgApi_1 = require("../QySendMsgApi");
/**
 * @author Javen
 * @copyright javendev@126.com
 * @description 任务卡片消息
 */
class QyTaskCardMsg extends QyBaseMsg_1.QyBaseMsg {
    constructor(taskCard, agentId, toUser, toParty, toTag, safe = 0, enableIdTrans = 0, enableDuplicateCheck = 0, duplicateCheckInterval = 1800) {
        super(QySendMsgApi_1.QySendMessageType.TASKCARD, agentId, toUser, toParty, toTag, safe, enableIdTrans, enableDuplicateCheck, duplicateCheckInterval);
        this.taskcard = taskCard;
    }
    get taskCard() {
        return this.taskcard;
    }
    set setText(taskCard) {
        this.taskcard = taskCard;
    }
}
exports.QyTaskCardMsg = QyTaskCardMsg;
class QyTaskCard {
    constructor(taskId, title, description, btn, url) {
        this.title = title;
        this.description = description;
        this.url = url;
        this.task_id = taskId;
        this.btn = btn;
    }
    set setTitle(title) {
        this.title = title;
    }
    get getTitle() {
        return this.title;
    }
    get getDescription() {
        return this.description;
    }
    set setDescription(description) {
        this.description = description;
    }
    set setUrl(url) {
        this.url = url;
    }
    get getUrl() {
        return this.url;
    }
    set setBtn(btn) {
        this.btn = btn;
    }
    get getBtn() {
        return this.btn;
    }
    set taskId(taskId) {
        this.task_id = taskId;
    }
    get taskId() {
        return this.task_id;
    }
}
exports.QyTaskCard = QyTaskCard;
class QyTaskCardBtn {
    constructor(key, name, replaceName, color, isBold) {
        this.key = key;
        this.name = name;
        this.replace_name = replaceName;
        this.color = color;
        this.is_bold = isBold;
    }
    set setKey(key) {
        this.key = key;
    }
    get getKey() {
        return this.key;
    }
    set setName(name) {
        this.name = name;
    }
    get getName() {
        return this.name;
    }
    set replaceName(replaceName) {
        this.replace_name = replaceName;
    }
    get replaceName() {
        return this.replace_name;
    }
    set setColor(color) {
        this.color = color;
    }
    get getColor() {
        return this.color;
    }
    set isBold(isBold) {
        this.is_bold = isBold;
    }
    get isBold() {
        return this.is_bold;
    }
}
exports.QyTaskCardBtn = QyTaskCardBtn;
