// @tnwx/wxcp/dist/entity/QyTextCardMsg.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const QyBaseMsg_1 = require("./QyBaseMsg");
const QySendMsgApi_1 = require("../QySendMsgApi");
/**
 * @author Javen
 * @copyright javendev@126.com
 * @description 文本卡片消息
 */
class QyTextCardMsg extends QyBaseMsg_1.QyBaseMsg {
    constructor(textCard, agentId, toUser, toParty, toTag, safe = 0, enableIdTrans = 0, enableDuplicateCheck = 0, duplicateCheckInterval = 1800) {
        super(QySendMsgApi_1.QySendMessageType.TEXTCARD, agentId, toUser, toParty, toTag, safe, enableIdTrans, enableDuplicateCheck, duplicateCheckInterval);
        this.textcard = textCard;
    }
    get textCard() {
        return this.textcard;
    }
    set textCard(textCard) {
        this.textcard = textCard;
    }
}
exports.QyTextCardMsg = QyTextCardMsg;
class QyTextCard {
    constructor(title, description, url, btnTxt) {
        this.title = title;
        this.description = description;
        this.url = url;
        this.btntxt = btnTxt;
    }
    set setTitle(title) {
        this.title = title;
    }
    get getTitle() {
        return this.title;
    }
    get getDescription() {
        return this.description;
    }
    set setDescription(description) {
        this.description = description;
    }
    set setUrl(url) {
        this.url = url;
    }
    get getUrl() {
        return this.url;
    }
    set setBtnTxt(btntxt) {
        this.btntxt = btntxt;
    }
    get getBtnTxt() {
        return this.btntxt;
    }
}
exports.QyTextCard = QyTextCard;
