// @tnwx/wxcp/dist/entity/QyTextMsg.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const QyBaseMsg_1 = require("./QyBaseMsg");
const QySendMsgApi_1 = require("../QySendMsgApi");
/**
 * @author Javen
 * @copyright javendev@126.com
 * @description 文本消息
 */
class QyTextMsg extends QyBaseMsg_1.QyBaseMsg {
    constructor(text, agentId, toUser, toParty, toTag, safe = 0, enableIdTrans = 0, enableDuplicateCheck = 0, duplicateCheckInterval = 1800) {
        super(QySendMsgApi_1.QySendMessageType.TEXT, agentId, toUser, toParty, toTag, safe, enableIdTrans, enableDuplicateCheck, duplicateCheckInterval);
        this.text = text;
    }
    get getText() {
        return this.text;
    }
    set setText(text) {
        this.text = text;
    }
}
exports.QyTextMsg = QyTextMsg;
class QyText {
    constructor(content, mentionedList, mentionedMobileList) {
        this.content = content;
        this.mentioned_list = mentionedList;
        this.mentioned_mobile_list = mentionedMobileList;
    }
    set setContent(content) {
        this.content = content;
    }
    get getContent() {
        return this.content;
    }
    set mentionedList(mentioned_list) {
        this.mentioned_list = mentioned_list;
    }
    get mentionedList() {
        return this.mentioned_list;
    }
    set mentionedMobileList(mentionedMobileList) {
        this.mentioned_mobile_list = mentionedMobileList;
    }
    get mentionedMobileList() {
        return this.mentioned_mobile_list;
    }
}
exports.QyText = QyText;
