// @tnwx/wxcp/dist/entity/QyVideoMsg.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const QyBaseMsg_1 = require("./QyBaseMsg");
const QySendMsgApi_1 = require("../QySendMsgApi");
/**
 * @author Javen
 * @copyright javendev@126.com
 * @description 视频消息
 */
class QyVideoMsg extends QyBaseMsg_1.QyBaseMsg {
    constructor(video, agentId, toUser, toParty, toTag, safe = 0, enableIdTrans = 0, enableDuplicateCheck = 0, duplicateCheckInterval = 1800) {
        super(QySendMsgApi_1.QySendMessageType.VIDEO, agentId, toUser, toParty, toTag, safe, enableIdTrans, enableDuplicateCheck, duplicateCheckInterval);
        this.video = video;
    }
    get getVideo() {
        return this.video;
    }
    set setVideo(video) {
        this.video = video;
    }
}
exports.QyVideoMsg = QyVideoMsg;
class QyVideo extends QyBaseMsg_1.QyMediaId {
    constructor(mediaId, title, description) {
        super(mediaId);
        this.title = title;
        this.description = description;
    }
    set setTitle(title) {
        this.title = title;
    }
    get getTitle() {
        return this.title;
    }
    get getDescription() {
        return this.description;
    }
    set setDescription(description) {
        this.description = description;
    }
}
exports.QyVideo = QyVideo;
