// @tnwx/wxmp/dist/api/shakearound/ShakeAroundApi.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
const util = require("util");
const accesstoken_1 = require("@tnwx/accesstoken");
const kits_1 = require("@tnwx/kits");
/**
 * @author Javen
 * @copyright javendev@126.com
 * @description 微信摇一摇
 */
class ShakeAroundApi {
    /**
     * 申请开通功能
     * @param name 联系人姓名，不超过20汉字或40个英文字母
     * @param phoneNumber 联系人电话
     * @param email 联系人邮箱
     * @param industryId 平台定义的行业代号
     * @param qualificationCertUrls 相关资质文件的图片url，图片需先上传至微信侧服务器，用“素材管理-上传图片素材”接口上传图片，返回的图片URL再配置在此处； 当不需要资质文件时，数组内可以不填写url
     * @param applyReason 申请理由，不超过250汉字或500个英文字母
     * @param accessToken
     */
    static register(name, phoneNumber, email, industryId, qualificationCertUrls, applyReason, accessToken) {
        return __awaiter(this, void 0, void 0, function* () {
            if (!accessToken) {
                accessToken = yield accesstoken_1.AccessTokenApi.getAccessToken();
            }
            let url = util.format(this.registerUrl, accessToken.getAccessToken);
            return kits_1.HttpKit.getHttpDelegate.httpPost(url, JSON.stringify({
                name: name,
                phone_number: phoneNumber,
                email: email,
                industry_id: industryId,
                qualification_cert_urls: qualificationCertUrls,
                apply_reason: applyReason || ''
            }));
        });
    }
    /**
     * 查询审核状态
     * @param accessToken
     */
    static auditStatus(accessToken) {
        return __awaiter(this, void 0, void 0, function* () {
            if (!accessToken) {
                accessToken = yield accesstoken_1.AccessTokenApi.getAccessToken();
            }
            let url = util.format(this.auditStatusUrl, accessToken.getAccessToken);
            return kits_1.HttpKit.getHttpDelegate.httpGet(url);
        });
    }
    /**
     * 上传图片素材
     * @param type Icon：摇一摇页面展示的icon图；License：申请开通摇一摇周边功能时需上传的资质文件；若不传type，则默认type=icon
     * @param filePath 文件路径 图片格式限定为：jpg,jpeg,png,gif。
     * @param accessToken
     */
    static addMaterial(type = MaterialType.icon, filePath, accessToken) {
        return __awaiter(this, void 0, void 0, function* () {
            if (!accessToken) {
                accessToken = yield accesstoken_1.AccessTokenApi.getAccessToken();
            }
            let url = util.format(this.addMaterialUrl, accessToken.getAccessToken);
            return kits_1.HttpKit.getHttpDelegate.upload(url, filePath, JSON.stringify({
                type: type
            }));
        });
    }
}
exports.ShakeAroundApi = ShakeAroundApi;
ShakeAroundApi.registerUrl = 'https://api.weixin.qq.com/shakearound/account/register?access_token=%s';
ShakeAroundApi.auditStatusUrl = 'https://api.weixin.qq.com/shakearound/account/auditstatus?access_token=%s';
ShakeAroundApi.addMaterialUrl = 'https://api.weixin.qq.com/shakearound/material/add?access_token=%s';
var MaterialType;
(function (MaterialType) {
    MaterialType["icon"] = "Icon";
    MaterialType["license"] = "License";
})(MaterialType = exports.MaterialType || (exports.MaterialType = {}));
