// @tnwx/wxpay/dist/WxApiType.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
/**
 * @author Javen
 * @copyright javendev@126.com
 * @description 微信支付相关 Api
 */
var WX_API_TYPE;
(function (WX_API_TYPE) {
    /**
     *  沙箱环境
     */
    WX_API_TYPE["SAND_BOX_NEW"] = "/sandboxnew";
    /**
     *  获取沙箱环境验签秘钥
     */
    WX_API_TYPE["GET_SIGN_KEY"] = "/sandboxnew/pay/getsignkey";
    /**
     *  统一下单
     */
    WX_API_TYPE["UNIFIED_ORDER"] = "/pay/unifiedorder";
    /**
     *  提交付款码支付
     */
    WX_API_TYPE["MICRO_PAY"] = "/pay/micropay";
    /**
     *  查询订单
     */
    WX_API_TYPE["ORDER_QUERY"] = "/pay/orderquery";
    /**
     *  关闭订单
     */
    WX_API_TYPE["CLOSE_ORDER"] = "/pay/closeorder";
    /**
     *  撤销订单
     */
    WX_API_TYPE["REVERSE"] = "/secapi/pay/reverse";
    /**
     *  申请退款
     */
    WX_API_TYPE["REFUND"] = "/secapi/pay/refund";
    /**
     *  查询退款
     */
    WX_API_TYPE["REFUND_QUERY"] = "/pay/refundquery";
    /**
     *  下载对账单
     */
    WX_API_TYPE["DOWNLOAD_BILL"] = "/pay/downloadbill";
    /**
     *  下载资金对账单
     */
    WX_API_TYPE["DOWNLOAD_FUND_FLOW"] = "/pay/downloadfundflow";
    /**
     *  交易保障
     */
    WX_API_TYPE["REPORT"] = "/payitil/report";
    /**
     *  转换短链接
     */
    WX_API_TYPE["SHORT_URL"] = "/tools/shorturl";
    /**
     *  授权码查询 openId
     */
    WX_API_TYPE["AUTH_CODE_TO_OPENID"] = "/tools/authcodetoopenid";
    /**
     *  拉取订单评价数据
     */
    WX_API_TYPE["BATCH_QUERY_COMMENT"] = "/billcommentsp/batchquerycomment";
    /**
     *  企业付款
     */
    WX_API_TYPE["TRANSFER"] = "/mmpaymkttransfers/promotion/transfers";
    /**
     *  查询企业付款
     */
    WX_API_TYPE["GET_TRANSFER_INFO"] = "/mmpaymkttransfers/gettransferinfo";
    /**
     *  企业付款到银行卡
     */
    WX_API_TYPE["TRANSFER_BANK"] = "/mmpaysptrans/pay_bank";
    /**
     *  查询企业付款到银行卡
     */
    WX_API_TYPE["GET_TRANSFER_BANK_INFO"] = "/mmpaysptrans/query_bank";
    /**
     *  发放企业红包
     */
    WX_API_TYPE["SEND_WORK_WX_REDPACK"] = "/mmpaymkttransfers/sendworkwxredpack";
    /**
     *  查询红包记录
     */
    WX_API_TYPE["QUERY_WORK_WX_REDPACK"] = "/mmpaymkttransfers/queryworkwxredpack";
    /**
     *  向员工付款
     */
    WX_API_TYPE["PAY_WWSP_TRANS_2_POCKET"] = "/mmpaymkttransfers/promotion/paywwsptrans2pocket";
    /**
     *  查询付款记录
     */
    WX_API_TYPE["QUERY_WWSP_TRANS_2_POCKET"] = "/mmpaymkttransfers/promotion/querywwsptrans2pocket";
    /**
     *  获取 RSA 加密公钥
     */
    WX_API_TYPE["GET_PUBLIC_KEY"] = "/risk/getpublickey";
    /**
     *  发放红包
     */
    WX_API_TYPE["SEND_RED_PACK"] = "/mmpaymkttransfers/sendredpack";
    /**
     *  发放裂变红包
     */
    WX_API_TYPE["SEND_GROUP_RED_PACK"] = "/mmpaymkttransfers/sendgroupredpack";
    /**
     *  查询红包记录
     */
    WX_API_TYPE["GET_HB_INFO"] = "/mmpaymkttransfers/gethbinfo";
    /**
     *  小程序发红包
     */
    WX_API_TYPE["SEND_MINI_PROGRAM_HB"] = "/mmpaymkttransfers/sendminiprogramhb";
    /**
     *  发放代金券
     */
    WX_API_TYPE["SEND_COUPON"] = "/mmpaymkttransfers/send_coupon";
    /**
     *  查询代金券批次
     */
    WX_API_TYPE["QUERY_COUPON_STOCK"] = "/mmpaymkttransfers/query_coupon_stock";
    /**
     *  查询代金券信息
     */
    WX_API_TYPE["QUERY_COUPONS_INFO"] = "/mmpaymkttransfers/querycouponsinfo";
    /**
     *  请求单次分账
     */
    WX_API_TYPE["PROFIT_SHARING"] = "/secapi/pay/profitsharing";
    /**
     *  请求多次分账
     */
    WX_API_TYPE["MULTI_PROFIT_SHARING"] = "/secapi/pay/multiprofitsharing";
    /**
     *  查询分账结果
     */
    WX_API_TYPE["PROFIT_SHARING_QUERY"] = "/pay/profitsharingquery";
    /**
     *  添加分账接收方
     */
    WX_API_TYPE["PROFITS_HARING_ADD_RECEIVER"] = "/pay/profitsharingaddreceiver";
    /**
     *  删除分账接收方
     */
    WX_API_TYPE["PROFIT_SHARING_REMOVE_RECEIVER"] = "/pay/profitsharingremovereceiver";
    /**
     *  完结分账
     */
    WX_API_TYPE["PROFIT_SHARING_FINISH"] = "/secapi/pay/profitsharingfinish";
    /**
     *  分账回退
     */
    WX_API_TYPE["PROFIT_SHARING_RETURN"] = "/secapi/pay/profitsharingreturn";
    /**
     *  分账回退结果查询
     */
    WX_API_TYPE["PROFIT_SHARING_RETURN_QUERY"] = "/pay/profitsharingreturnquery";
    /**
     *  支付押金（人脸支付）
     */
    WX_API_TYPE["DEPOSIT_FACE_PAY"] = "/deposit/facepay";
    /**
     *  支付押金（付款码支付）
     */
    WX_API_TYPE["DEPOSIT_MICRO_PAY"] = "/deposit/micropay";
    /**
     *  查询订单（押金）
     */
    WX_API_TYPE["DEPOSIT_ORDER_QUERY"] = "/deposit/orderquery";
    /**
     *  撤销订单（押金）
     */
    WX_API_TYPE["DEPOSIT_REVERSE"] = "/deposit/reverse";
    /**
     *  消费押金
     */
    WX_API_TYPE["DEPOSIT_CONSUME"] = "/deposit/consume";
    /**
     *  申请退款（押金）
     */
    WX_API_TYPE["DEPOSIT_REFUND"] = "/deposit/refund";
    /**
     *  查询退款（押金）
     */
    WX_API_TYPE["DEPOSIT_REFUND_QUERY"] = "deposit/refundquery";
    /**
     *  公众号纯签约
     */
    WX_API_TYPE["ENTRUST_WEB"] = "/papay/entrustweb";
    /**
     *  公众号纯签约（服务商模式）
     */
    WX_API_TYPE["PARTNER_ENTRUST_WEB"] = "/papay/partner/entrustweb";
    /**
     *  APP纯签约
     */
    WX_API_TYPE["PRE_ENTRUST_WEB"] = "/papay/preentrustweb";
    /**
     *  APP纯签约（服务商模式）
     */
    WX_API_TYPE["PARTNER_PRE_ENTRUST_WEB"] = "/papay/partner/preentrustweb";
    /**
     *  H5纯签约
     */
    WX_API_TYPE["H5_ENTRUST_WEB"] = "/papay/h5entrustweb";
    /**
     *  H5纯签约（服务商模式）
     */
    WX_API_TYPE["PARTNER_H5_ENTRUST_WEB"] = "/papay/partner/h5entrustweb";
    /**
     *  支付中签约
     */
    WX_API_TYPE["PAY_CONTRACT_ORDER"] = "/pay/contractorder";
    /**
     *  查询签约关系
     */
    WX_API_TYPE["QUERY_ENTRUST_CONTRACT"] = "/papay/querycontract";
    /**
     *  查询签约关系（服务商模式）
     */
    WX_API_TYPE["PARTNER_QUERY_ENTRUST_CONTRACT"] = "/papay/partner/querycontract";
    /**
     *  代扣申请扣款
     */
    WX_API_TYPE["PAP_PAY_APPLY"] = "/pay/pappayapply";
    /**
     *  代扣申请扣款（服务商模式）
     */
    WX_API_TYPE["PARTNER_PAP_PAY_APPLY"] = "/pay/partner/pappayapply";
    /**
     *  查询代扣订单
     */
    WX_API_TYPE["PAP_ORDER_QUERY"] = "/pay/paporderquery";
    /**
     *  查询代扣订单
     */
    WX_API_TYPE["PARTNER_PAP_ORDER_QUERY"] = "/pay/partner/paporderquery";
    /**
     *  代扣申请解约
     */
    WX_API_TYPE["DELETE_ENTRUST_CONTRACT"] = "/papay/deletecontract";
    /**
     *  代扣申请解约（服务商模式）
     */
    WX_API_TYPE["PARTNER_DELETE_ENTRUST_CONTRACT"] = "/papay/partner/deletecontract";
    /**
     *  刷脸支付
     */
    WX_API_TYPE["FACE_PAY"] = "/pay/facepay";
    /**
     *  查询刷脸支付订单
     */
    WX_API_TYPE["FACE_PAY_QUERY"] = "/pay/facepayqueryy";
    /**
     *  撤销刷脸支付订单
     */
    WX_API_TYPE["FACE_PAY_REVERSE"] = "/secapi/pay/facepayreverse";
    /**
     *  小微商户申请入驻
     */
    WX_API_TYPE["MICRO_SUBMIT"] = "/applyment/micro/submit";
    /**
     *  查询申请状态
     */
    WX_API_TYPE["GET_MICRO_SUBMIT_STATE"] = "/applyment/micro/getstate";
    /**
     *  提交升级申请
     */
    WX_API_TYPE["MICRO_SUBMIT_UPGRADE"] = "/applyment/micro/submitupgrade";
    /**
     *  查询升级申请单状态
     */
    WX_API_TYPE["GET_MICRO_UPGRADE_STATE"] = "/applyment/micro/getupgradestate";
    /**
     *  查询提现状态
     */
    WX_API_TYPE["QUERY_AUTO_WITH_DRAW_BY_DATE"] = "/fund/queryautowithdrawbydate";
    /**
     *  修改结算银行卡
     */
    WX_API_TYPE["MICRO_MODIFY_ARCHIVES"] = "/applyment/micro/modifyarchives";
    /**
     *  重新发起提现
     */
    WX_API_TYPE["RE_AUTO_WITH_DRAW_BY_DATE"] = "/fund/reautowithdrawbydate";
    /**
     *  修改联系信息
     */
    WX_API_TYPE["MICRO_MODIFY_CONTACT_INFO"] = "/applyment/micro/modifycontactinfo";
    /**
     *  小微商户关注功能配置
     */
    WX_API_TYPE["ADD_RECOMMEND_CONF"] = "/secapi/mkt/addrecommendconf";
    /**
     *  小微商户开发配置新增支付目录
     */
    WX_API_TYPE["ADD_SUB_DEV_CONFIG"] = "/secapi/mch/addsubdevconfig";
    /**
     *  小微商户开发配置查询
     */
    WX_API_TYPE["QUERY_SUB_DEV_CONFIG"] = "/secapi/mch/querysubdevconfig";
    /**
     * 小微商户图片上传
     */
    WX_API_TYPE["MCH_UPLOAD_MEDIA"] = "/secapi/mch/uploadmedia";
    /**
     * 获取平台证书列表
     */
    WX_API_TYPE["GET_CERTIFICATES"] = "/v3/certificates";
    /**
     * 营销专用-图片上传
     */
    WX_API_TYPE["MARKETING_UPLOAD_MEDIA"] = "/v3/marketing/favor/media/image-upload";
    /**
     * 通用接口-图片上传
     */
    WX_API_TYPE["MERCHANT_UPLOAD_MEDIA"] = "/v3/merchant/media/upload";
    /**
     * 通用接口-视频上传
     */
    WX_API_TYPE["MERCHANT_VIDEO_UPLOAD"] = "/v3/merchant/media/video_upload";
    /**
     * 创建/查询支付分订单
     */
    WX_API_TYPE["PAY_SCORE_SERVICE_ORDER"] = "/v3/payscore/serviceorder";
    /**
     * 取消支付分订单
     */
    WX_API_TYPE["PAY_SCORE_SERVICE_ORDER_CANCEL"] = "/v3/payscore/serviceorder/%s/cancel";
    /**
     * 修改支付分订单金额
     */
    WX_API_TYPE["PAY_SCORE_SERVICE_ORDER_MODIFY"] = "/v3/payscore/serviceorder/%s/modify";
    /**
     * 完结支付分订单
     */
    WX_API_TYPE["PAY_SCORE_SERVICE_ORDER_COMPLETE"] = "/v3/payscore/serviceorder/%s/complete";
    /**
     * 支付分订单收款
     */
    WX_API_TYPE["PAY_SCORE_SERVICE_ORDER_PAY"] = "/v3/payscore/serviceorder/%s/pay";
    /**
     * 同步服务订单信息
     */
    WX_API_TYPE["PAY_SCORE_SERVICE_ORDER_SYNC"] = "/v3/payscore/payafter-orders/%s/sync";
    /**
     * 查询用户支付分开启状态
     */
    WX_API_TYPE["PAY_SCORE_USER_SERVICE_STATE"] = "/v3/payscore/user-service-state";
    /**
     * 特约商户进件-提交申请单
     */
    WX_API_TYPE["APPLY_4_SUB"] = "/v3/applyment4sub/applyment";
    /**
     * 特约商户进件-查询申请单状态
     */
    WX_API_TYPE["GET_APPLY_STATE"] = "/v3/applyment4sub/applyment/business_code/%s";
    /**
     * 特约商户进件-修改结算帐号
     */
    WX_API_TYPE["MODIFY_SETTLEMENT"] = "/v3/apply4sub/sub_merchants/%s/modify-settlement";
    /**
     * 特约商户进件-查询结算账户
     */
    WX_API_TYPE["GET_SETTLEMENT"] = "/v3/apply4sub/sub_merchants/%s/settlement";
    /**
     * 商户开户意愿确认-提交申请单 OR 查询申请单审核结果
     */
    WX_API_TYPE["MER_OPEN_APPLY_SUBMIT_OR_RESULT"] = "/v3/apply4subject/applyment";
    /**
     * 商户开户意愿确认-撤销申请单
     */
    WX_API_TYPE["MER_OPEN_APPLY_CANCEL"] = "/v3/apply4subject/applyment/%s/cancel";
    /**
     * 商户开户意愿确认-获取商户开户意愿确认状态
     */
    WX_API_TYPE["GET_MER_OPEN_APPLY_STATE"] = "/v3/apply4subject/applyment/merchants/%s/state";
    /**
     * 商业支付投诉-查询投诉信息
     */
    WX_API_TYPE["MERCHANT_SERVICE_COMPLAINTS"] = "/v3/merchant-service/complaints";
    /**
     * 商业支付投诉-创建/查询/更新/删除投诉通知回调
     */
    WX_API_TYPE["MERCHANT_SERVICE_COMPLAINTS_NOTIFICATIONS"] = "/v3/merchant-service/complaint-notifications";
    /**
     * 代金券-创建代金券批次
     */
    WX_API_TYPE["CREATE_COUPON_STOCKS"] = "/v3/marketing/favor/coupon-stocks";
    /**
     * 代金券-激活代金券批次
     */
    WX_API_TYPE["START_COUPON_STOCKS"] = "/v3/marketing/favor/stocks/%s/start";
    /**
     * 代金券-发放代金券
     */
    WX_API_TYPE["COUPON_SEND"] = "/v3/marketing/favor/users/%s/coupons";
    /**
     * 代金券-暂停代金券批次
     */
    WX_API_TYPE["PAUSE_COUPON_STOCKS"] = "/v3/marketing/favor/stocks/%s/pause";
    /**
     * 代金券-重启代金券批次
     */
    WX_API_TYPE["RESTART_COUPON_STOCKS"] = "/v3/marketing/favor/stocks/%s/restart";
    /**
     * 代金券-条件查询批次列表
     */
    WX_API_TYPE["QUERY_COUPON_STOCKS"] = "/v3/marketing/favor/stocks";
    /**
     * 代金券-查询批次详情
     */
    WX_API_TYPE["QUERY_COUPON_STOCKS_INFO"] = "/v3/marketing/favor/stocks/%s";
    /**
     * 代金券-查询代金券详情
     */
    WX_API_TYPE["QUERY_COUPON_INFO"] = "/v3/marketing/favor/users/%s/coupons/%s";
    /**
     * 代金券-查询代金券可用商户
     */
    WX_API_TYPE["QUERY_COUPON_MERCHANTS"] = "/v3/marketing/favor/stocks/%s/merchants";
    /**
     * 代金券-查询代金券可用单品
     */
    WX_API_TYPE["QUERY_COUPON_ITEMS"] = "/v3/marketing/favor/stocks/%s/items";
    /**
     * 代金券-根据商户号查用户的券
     */
    WX_API_TYPE["QUERY_USER_COUPON"] = "/v3/marketing/favor/users/%s/coupons";
    /**
     * 代金券-下载批次核销明细
     */
    WX_API_TYPE["COUPON_STOCKS_USER_FLOW_DOWNLOAD"] = "/v3/marketing/favor/stocks/%s/use-flow";
    /**
     * 代金券-下载批次退款明细
     */
    WX_API_TYPE["COUPON_STOCKS_REFUND_FLOW_DOWNLOAD"] = "/v3/marketing/favor/stocks/%s/refund-flow";
    /**
     * 代金券-设置消息通知地址
     */
    WX_API_TYPE["SETTING_COUPON_CALLBACKS"] = "/v3/marketing/favor/callbacks";
    /**
     * 商家券-创建商家券
     */
    WX_API_TYPE["CREATE_BUSINESS_COUPON"] = "/v3/marketing/busifavor/stocks";
    /**
     * 发放消费卡
     */
    WX_API_TYPE["SEND_BUSINESS_COUPON"] = "/v3/marketing/busifavor/coupons/%s/send";
    /**
     * H5 发券
     */
    WX_API_TYPE["H5_SEND_COUPON"] = "/busifavor/getcouponinfo";
    /**
     * 商家券-查询商家券批次详情
     */
    WX_API_TYPE["QUERY_BUSINESS_COUPON_STOCKS_INFO"] = "/v3/marketing/busifavor/stocks/%s";
    /**
     * 商家券-查询商家券批次详情
     */
    WX_API_TYPE["USE_BUSINESS_COUPON"] = "/v3/marketing/busifavor/coupons/use";
    /**
     * 商家券-根据过滤条件查询用户券
     */
    WX_API_TYPE["QUERY_BUSINESS_USER_COUPON"] = "/v3/marketing/busifavor/users/%s/coupons";
    /**
     * 商家券-查询用户单张券详情
     */
    WX_API_TYPE["QUERY_BUSINESS_USER_COUPON_INFO"] = "/v3/marketing/busifavor/users/%s/coupons/%s/appids/%s";
    /**
     * 商家券-上传预存code
     */
    WX_API_TYPE["BUSINESS_COUPON_UPLOAD_CODE"] = "/v3/marketing/busifavor/stocks/%/couponcodes";
    /**
     * 商家券-设置/查询商家券事件通知地址
     */
    WX_API_TYPE["BUSINESS_COUPON_CALLBACKS"] = "/v3/marketing/busifavor/callbacks";
    /**
     * 关联订单信息
     */
    WX_API_TYPE["BUSINESS_COUPON_ASSOCIATE"] = "/v3/marketing/busifavor/coupons/associate";
    /**
     * 取消关联订单信息
     */
    WX_API_TYPE["BUSINESS_COUPON_DISASSOCIATE"] = "/v3/marketing/busifavor/coupons/disassociate";
    /**
     * 支付有礼-创建全场满额送活动
     */
    WX_API_TYPE["PAY_GIFT_ACTIVITY"] = "/v3/marketing/paygiftactivity/unique-threshold-activity";
    /**
     * 支付有礼-查询活动详情接口
     */
    WX_API_TYPE["PAY_GIFT_ACTIVITY_INFO"] = "/v3/marketing/paygiftactivity/activities/%s";
    /**
     * 支付有礼-获取支付有礼活动列表
     */
    WX_API_TYPE["PAY_GIFT_ACTIVITY_GET"] = "/v3/marketing/paygiftactivity/activities";
    /**
     * 支付有礼-查询活动发券商户号
     */
    WX_API_TYPE["PAY_GIFT_ACTIVITY_QUERY_MER"] = "/v3/marketing/paygiftactivity/activities/%s/merchants";
    /**
     * 支付有礼-查询活动指定商品列表
     */
    WX_API_TYPE["PAY_GIFT_ACTIVITY_QUERY_GOODS"] = "/v3/marketing/paygiftactivity/activities/%s/goods";
    /**
     * 支付有礼-终止活动
     */
    WX_API_TYPE["PAY_GIFT_ACTIVITY_TERMINATE"] = "/v3/marketing/paygiftactivity/activities/%s/terminate";
    /**
     * 支付有礼-新增活动发券商户号
     */
    WX_API_TYPE["PAY_GIFT_ACTIVITY_ADD_MERCHANTS"] = "/v3/marketing/paygiftactivity/activities/%s/merchants/add";
    /**
     * 支付有礼-删除活动发券商户号
     */
    WX_API_TYPE["PAY_GIFT_ACTIVITY_DELETE_MERCHANTS"] = "/v3/marketing/paygiftactivity/activities/%s/merchants/delete";
    /**
     * 点金计划-特约商户管理
     */
    WX_API_TYPE["CHANGE_GOLD_PLAN_STATUS"] = "/v3/goldplan/merchants/changegoldplanstatus";
    /**
     * 点金计划-商家小票管理
     */
    WX_API_TYPE["CHANGE_CUSTOM_PAGE_STATUS"] = "/v3/goldplan/merchants/changecustompagestatus";
    /**
     * 点金计划-同业过滤标签管理
     */
    WX_API_TYPE["SET_ADVERTISING_INDUSTRY_FILTER"] = "/v3/goldplan/merchants/set-advertising-industry-filter";
    /**
     * 电商收付通-二级商户进件
     */
    WX_API_TYPE["E_COMMERCE_APPLY"] = "/v3/ecommerce/applyments";
    /**
     * 电商收付通-查询进件申请状态
     */
    WX_API_TYPE["E_COMMERCE_APPLY_STATE"] = "/v3/ecommerce/applyments/%s";
    /**
     * 电商收付通-通过业务申请编号查询申请状态
     */
    WX_API_TYPE["E_COMMERCE_APPLY_STATE_BY_NO"] = "/v3/ecommerce/applyments/out-request-no/%s";
    /**
     * 合单下单-APP支付
     */
    WX_API_TYPE["COMBINE_TRANSACTIONS_APP"] = "/v3/combine-transactions/app";
    /**
     * 合单下单-JS支付
     */
    WX_API_TYPE["COMBINE_TRANSACTIONS_JS"] = "/v3/combine-transactions/jsapi";
    /**
     * 合单下单-H5支付
     */
    WX_API_TYPE["COMBINE_TRANSACTIONS_H5"] = "/v3/combine-transactions/h5";
    /**
     * 合单下单-Native支付
     */
    WX_API_TYPE["COMBINE_TRANSACTIONS_NATIVE"] = "/v3/combine-transactions/native";
    /**
     * 合单下单-合单查询订单
     */
    WX_API_TYPE["COMBINE_TRANSACTIONS_QUERY"] = "/v3/combine-transactions/out-trade-no/%s";
    /**
     * 合单下单-合单关闭订单
     */
    WX_API_TYPE["COMBINE_TRANSACTIONS_CLOSE"] = "/v3/combine-transactions/out-trade-no/%s/close";
    /**
     * 电商收付通-补差接口-请求补差
     */
    WX_API_TYPE["CREATE_SUBSIDIES"] = "v3/ecommerce/subsidies/create";
    /**
     * 电商收付通-补差接口-请求补差回退
     */
    WX_API_TYPE["RETURN_SUBSIDIES"] = "/v3/ecommerce/subsidies/return";
    /**
     * 电商收付通-补差接口-取消补差
     */
    WX_API_TYPE["CANCEL_SUBSIDIES"] = "/v3/ecommerce/subsidies/cancel";
    /**
     * 电商收付通-分账接口-请求分账/查询分账结果
     */
    WX_API_TYPE["PROFIT_SHARING_ORDERS"] = "/v3/ecommerce/profitsharing/orders";
    /**
     * 电商收付通-分账接口-查询分账回退结果
     */
    WX_API_TYPE["PROFIT_SHARING_RETURN_ORDERS"] = "/v3/ecommerce/profitsharing/returnorders";
    /**
     * 电商收付通-分账接口-完结分账
     */
    WX_API_TYPE["PROFIT_SHARING_FINISH_ORDER"] = "/v3/ecommerce/profitsharing/finish-order";
    /**
     * 添加分账接收方
     */
    WX_API_TYPE["PROFIT_SHARING_RECEIVERS_ADD"] = "/v3/ecommerce/profitsharing/receivers/add";
    /**
     * 删除分账接收方
     */
    WX_API_TYPE["PROFIT_SHARING_RECEIVERS_DELETE"] = "/v3/ecommerce/profitsharing/receivers/delete";
    /**
     * 电商收付通-退款接口-退款申请
     */
    WX_API_TYPE["E_COMMERCE_REFUNDS"] = "/v3/ecommerce/refunds/apply";
    /**
     * 电商收付通-退款接口-通过微信支付退款单号查询退款
     */
    WX_API_TYPE["QUERY_REFUND"] = "/v3/ecommerce/refunds/id/%s";
    /**
     * 电商收付通-退款接口-通过商户退款单号查询退款
     */
    WX_API_TYPE["QUERY_REFUNDS_BY_REFUND_NO"] = "/v3/ecommerce/refunds/out-refund-no/%s";
    /**
     * 电商收付通-余额查询接口
     */
    WX_API_TYPE["QUERY_BALANCE"] = "/v3/ecommerce/fund/balance/%s";
    /**
     * 查询二级商户账户日终余额
     */
    WX_API_TYPE["QUERY_END_DAY_BALANCE"] = "/v3/ecommerce/fund/enddaybalance/%s";
    /**
     * 查询电商平台账户实时余额
     */
    WX_API_TYPE["QUERY_MERCHANT_BALANCE"] = "/v3/merchant/fund/balance/%s";
    /**
     * 查询电商平台账户日终余额
     */
    WX_API_TYPE["QUERY_MERCHANT_END_DAY_BALANCE"] = "/v3/merchant/fund/dayendbalance/%s";
    /**
     * 电商收付通-提现接口-账户余额提现
     */
    WX_API_TYPE["WITHDRAW"] = "/v3/ecommerce/fund/withdraw";
    /**
     * 电商收付通-提现接口-提现状态查询
     */
    WX_API_TYPE["QUERY_WITHDRAW"] = "/v3/ecommerce/fund/withdraw/%s";
    /**
     * 电商收付通-提现接口-商户提现单号查询
     */
    WX_API_TYPE["QUERY_WITHDRAW_BY_OUT_REQUEST_NO"] = "/v3/ecommerce/fund/withdraw/out-request-no/%s";
    /**
     * 电商收付通-提现接口-电商平台提现
     */
    WX_API_TYPE["MERCHANT_WITHDRAW"] = "/v3/merchant/fund/withdraw";
    /**
     * 电商收付通-提现接口-微信支付提现单号查询
     */
    WX_API_TYPE["QUERY_MERCHANT_WITHDRAW"] = "/v3/ecommerce/fund/withdraw/%s";
    /**
     * 电商收付通-提现接口-商户提现单号查询
     */
    WX_API_TYPE["QUERY_MERCHANT_WITHDRAW_BY_OUT_REQUEST_NO"] = "/v3/merchant/fund/withdraw/out-request-no/%s";
    /**
     * 电商收付通-提现接口-按日下载提现异常文件
     */
    WX_API_TYPE["WITHDRAW_BILL"] = "/v3/merchant/fund/withdraw/bill-type/%s";
    /**
     * 申请交易账单
     */
    WX_API_TYPE["TRADE_BILL"] = "/v3/bill/tradebill";
    /**
     * 申请资金账单
     */
    WX_API_TYPE["FUND_FLOW_BILL"] = "/v3/bill/fundflowbill";
    /**
     * 下载账单
     */
    WX_API_TYPE["BILL_DOWNLOAD"] = "/v3/billdownload/file";
    /**
     * 银行特约商户违规信息查询
     */
    WX_API_TYPE["GET_VIOLATION"] = "/risk/getviolation";
    /**
     * 事前-风险商户核查接口
     */
    WX_API_TYPE["QUERY_MCH_RISK"] = "/mchrisk/querymchrisk";
    /**
     * 事后-风险商户处理结果同步接口
     */
    WX_API_TYPE["SYNC_MCH_RISK_RESULT"] = "/mchrisk/syncmchriskresult";
    /**
     * 间联模式查询商户审核状态开放接口
     */
    WX_API_TYPE["BANK_QUERY_MCH_AUDIT_INFO"] = "/mchrisk/bankquerymchauditinfo";
    /**
     * 渠道商查询商户审核信息
     */
    WX_API_TYPE["CHANNEL_QUERY_MCH_AUDIT_INFO"] = "/mchrisk/channelquerymchauditinfo";
    /**
     * 设置风险通知回调链接
     */
    WX_API_TYPE["SET_MCH_RISK_CALLBACK"] = "/mchrisk/setmchriskcallback";
    /**
     * 向员工付款
     */
    WX_API_TYPE["PAY_WWS_TRANS_2_POCKET"] = "/mmpaymkttransfers/promotion/paywwsptrans2pocket";
    /**
     * 查询向员工付款记录
     */
    WX_API_TYPE["QUERY_WWS_TRANS_2_POCKET"] = "/mmpaymkttransfers/promotion/querywwsptrans2pocket";
    /**
     * 发放企业红包
     */
    WX_API_TYPE["SEND_WORK_WX_RED_PACK"] = "/mmpaymkttransfers/sendworkwxredpack";
    /**
     * 查询企业红包记录
     */
    WX_API_TYPE["QUERY_WORK_WX_RED_PACK"] = "/mmpaymkttransfers/queryworkwxredpack";
    /**
     * 查询/更新先享卡订单
     */
    WX_API_TYPE["DISCOUNT_CARD_ORDER"] = "/v3/discount-card/orders/%s";
    /**
     * 查询先享卡订单
     */
    WX_API_TYPE["DISCOUNT_CARD_ORDER_TRADE_NO"] = "/v3/discount-card/orders/out-trade-no/%s";
    /**
     * 服务人员注册
     */
    WX_API_TYPE["SMART_GUIDE_GUIDES"] = "/v3/smartguide/guides";
    /**
     * 服务人员分配
     */
    WX_API_TYPE["SMART_GUIDE_GUIDES_ASSIGN"] = "/v3/smartguide/guides/%s/assign";
    /**
     * 服务人员信息更新
     */
    WX_API_TYPE["SMART_GUIDE_GUIDES_UPDATE"] = "/v3/smartguide/guides/%s";
    /**
     * 报关接口-订单附加信息提交接口
     */
    WX_API_TYPE["CUSTOM_DECLARE_ORDER"] = "/cgi-bin/mch/customs/customdeclareorder";
    /**
     * 报关接口-订单附加信息查询接口
     */
    WX_API_TYPE["CUSTOM_DECLARE_QUERY"] = "/cgi-bin/mch/customs/customdeclarequery";
    /**
     * 报关接口-订单附加信息重推接口
     */
    WX_API_TYPE["CUSTOM_DECLARE_RE_DECLARE"] = "/cgi-bin/mch/newcustoms/customdeclareredeclare";
    /**
     * APP 下单 API
     */
    WX_API_TYPE["APP_PAY"] = "/v3/pay/transactions/app";
    WX_API_TYPE["PARTNER_APP_PAY"] = "/v3/pay/partner/transactions/app";
    /**
     * JS API 下单 API
     */
    WX_API_TYPE["JS_API_PAY"] = "/v3/pay/transactions/jsapi";
    WX_API_TYPE["PARTNER_JS_API_PAY"] = "/v3/pay/partner/transactions/jsapi";
    /**
     * Native 下单 API
     */
    WX_API_TYPE["NATIVE_PAY"] = "/v3/pay/transactions/native";
    WX_API_TYPE["PARTNER_NATIVE_PAY"] = "/v3/pay/partner/transactions/native";
    /**
     * H5 下单 API
     */
    WX_API_TYPE["H5_PAY"] = "/v3/pay/transactions/h5";
    WX_API_TYPE["PARTNER_H5_PAY"] = "/v3/pay/partner/transactions/h5";
    /**
     * 微信支付订单号查询
     */
    WX_API_TYPE["ORDER_QUERY_BY_ID"] = "/v3/pay/transactions/id/%s";
    WX_API_TYPE["PARTNER_ORDER_QUERY_BY_ID"] = "/v3/pay/partner/transactions/id/%s";
    /**
     * 商户订单号查询
     */
    WX_API_TYPE["ORDER_QUERY_BY_NO"] = "/v3/pay/transactions/out-trade-no/%s";
    WX_API_TYPE["PARTNER_ORDER_QUERY_BY_NO"] = "/v3/pay/partner/transactions/out-trade-no/%s";
    /**
     * 关闭订单
     */
    WX_API_TYPE["CLOSE_ORDER_BY_NO"] = "/v3/pay/transactions/out-trade-no/%s/close";
    WX_API_TYPE["PARTNER_CLOSE_ORDER_BY_NO"] = "/v3/pay/partner/transactions/out-trade-no/%s/close";
    /**
     * 申请退款
     */
    WX_API_TYPE["DOMESTIC_REFUNDS"] = "/v3/refund/domestic/refunds";
    /**
     * 查询单笔退款
     */
    WX_API_TYPE["DOMESTIC_REFUNDS_QUERY"] = "/v3/refund/domestic/refunds/%s";
    /**
     * 委托营销-建立合作关系
     */
    WX_API_TYPE["PARTNERSHIPS_BUILD"] = "/v3/marketing/partnerships/build";
    /**
     * 委托营销-终止合作关系
     */
    WX_API_TYPE["PARTNERSHIPS_TERMINATE"] = "/v3/marketing/partnerships/terminate";
    /**
     * 智慧商圈-商圈积分同步
     */
    WX_API_TYPE["BUSINESS_CIRCLE_POINTS_NOTIFY"] = "/v3/businesscircle/points/notify";
    /**
     * 连锁品牌-分账-查询分账
     */
    WX_API_TYPE["BRAND_PROFIT_SHARING_ORDERS"] = "/v3/brand/profitsharing/orders";
    /**
     * 连锁品牌-分账回退-查询分账回退
     */
    WX_API_TYPE["BRAND_PROFIT_SHARING_RETURN_ORDERS"] = "/v3/brand/profitsharing/returnorders";
    /**
     * 连锁品牌-完结分账
     */
    WX_API_TYPE["BRAND_PROFIT_SHARING_FINISH_ORDER"] = "/v3/brand/profitsharing/finish-order";
    /**
     * 连锁品牌-添加分账接收方
     */
    WX_API_TYPE["BRAND_PROFIT_SHARING_RECEIVERS_ADD"] = "/v3/brand/profitsharing/receivers/add";
    /**
     * 连锁品牌-删除分账接收方
     */
    WX_API_TYPE["BRAND_PROFIT_SHARING_RECEIVERS_delete"] = "/v3/brand/profitsharing/receivers/delete";
})(WX_API_TYPE = exports.WX_API_TYPE || (exports.WX_API_TYPE = {}));
