// @tnwx/accesstoken/dist/openmp/OpenAuthorizerAccessTokenApi.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
const kits_1 = require("@tnwx/kits");
const AccessToken_1 = require("../AccessToken");
const ApiConfigKit_1 = require("../ApiConfigKit");
const util = require("util");
/**
 * @author Javen
 * @copyright javendev@126.com
 * @description 微信开放平台第三方应用凭证 AccessToken
 */
class OpenAuthorizerAccessTokenApi {
    /**
     * 获取 acces_token
     * 1、先从缓存中获取，如果可用就直接返回
     * 2、如果缓存中的已过期就调用刷新接口来获取新的 acces_token
     *
     * componentAccessToken authorizerRefreshToken 为空时从缓存中获取
     *
     * @param authorizerAppId         授权方 appid
     * @param componentAccessToken    开放平台令牌
     * @param authorizerRefreshToken  刷新令牌
     */
    static getAccessToken(authorizerAppId, componentAccessToken, authorizerRefreshToken) {
        return __awaiter(this, void 0, void 0, function* () {
            let ac = ApiConfigKit_1.ApiConfigKit.getApiConfig;
            let accessToken = yield this.getAvailableAccessToken(ac, authorizerAppId);
            if (accessToken) {
                if (ApiConfigKit_1.ApiConfigKit.isDevMode()) {
                    console.debug('缓存中的 accesstoken');
                }
                return accessToken;
            }
            if (ApiConfigKit_1.ApiConfigKit.isDevMode()) {
                console.debug('刷新 accesstoken');
            }
            return yield this.refreshAccessToken(ac, componentAccessToken, authorizerAppId, authorizerRefreshToken);
        });
    }
    /**
     * 获取可用的 AccessToken
     * @param apiConfig
     * @param authorizerAppId
     */
    static getAvailableAccessToken(apiConfig, authorizerAppId) {
        return __awaiter(this, void 0, void 0, function* () {
            let result;
            let cache = ApiConfigKit_1.ApiConfigKit.getCache;
            let accessTokenJson = yield cache.get(apiConfig.getAppId.concat('_').concat(authorizerAppId));
            if (accessTokenJson) {
                result = new AccessToken_1.AccessToken(accessTokenJson, AccessToken_1.AccessTokenType.AUTHORIZER_TOKEN);
            }
            if (result && result.isAvailable()) {
                return result;
            }
            else {
                return undefined;
            }
        });
    }
    /**
     * 获取新的 acces_token 并设置缓存
     * @param apiConfig
     * @param componentAccessToken
     * @param authorizerAppId
     * @param authorizerRefreshToken
     */
    static refreshAccessToken(apiConfig, componentAccessToken, authorizerAppId, authorizerRefreshToken) {
        return __awaiter(this, void 0, void 0, function* () {
            let url = util.format(this.url, componentAccessToken);
            let data = yield kits_1.HttpKit.getHttpDelegate.httpPost(url, JSON.stringify({
                component_appid: apiConfig.getAppId,
                authorizer_appid: authorizerAppId,
                authorizer_refresh_token: authorizerRefreshToken
            }));
            if (data) {
                data = JSON.stringify(data);
                let accessToken = new AccessToken_1.AccessToken(data, AccessToken_1.AccessTokenType.AUTHORIZER_TOKEN);
                let cache = ApiConfigKit_1.ApiConfigKit.getCache;
                cache.set(apiConfig.getAppId.concat('_').concat(authorizerAppId), accessToken.getCacheJson);
                return accessToken;
            }
            else {
                throw new Error('获取 accessToken 异常');
            }
        });
    }
}
exports.OpenAuthorizerAccessTokenApi = OpenAuthorizerAccessTokenApi;
OpenAuthorizerAccessTokenApi.url = 'https://api.weixin.qq.com/cgi-bin/component/api_authorizer_token?component_access_token=%s';
