// @tnwx/accesstoken/dist/openmp/OpenComponentAccessTokenApi.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
const kits_1 = require("@tnwx/kits");
const AccessToken_1 = require("../AccessToken");
const ApiConfigKit_1 = require("../ApiConfigKit");
/**
 * @author Javen
 * @copyright javendev@126.com
 * @description 微信开发平台 AccessToken
 */
class OpenComponentAccessTokenApi {
    /**
     * 获取 acces_token
     * 1、先从缓存中获取，如果可用就直接返回
     * 2、如果缓存中的已过期就调用刷新接口来获取新的 acces_token
     */
    static getAccessToken() {
        return __awaiter(this, void 0, void 0, function* () {
            let ac = ApiConfigKit_1.ApiConfigKit.getApiConfig;
            let accessToken = yield this.getAvailableAccessToken(ac);
            if (accessToken) {
                if (ApiConfigKit_1.ApiConfigKit.isDevMode()) {
                    console.debug('缓存中的 accesstoken');
                }
                return accessToken;
            }
            if (ApiConfigKit_1.ApiConfigKit.isDevMode()) {
                console.debug('刷新 accesstoken');
            }
            return yield this.refreshAccessToken(ac);
        });
    }
    /**
     * 获取可用的 AccessToken
     * @param apiConfig
     */
    static getAvailableAccessToken(apiConfig) {
        return __awaiter(this, void 0, void 0, function* () {
            let result;
            let cache = ApiConfigKit_1.ApiConfigKit.getCache;
            let accessTokenJson = yield cache.get(apiConfig.getAppId);
            if (accessTokenJson) {
                result = new AccessToken_1.AccessToken(accessTokenJson, AccessToken_1.AccessTokenType.COMPONENT_TOKEN);
            }
            if (result && result.isAvailable()) {
                return result;
            }
            else {
                return undefined;
            }
        });
    }
    /**
     * 获取新的 acces_token 并设置缓存
     * @param apiConfig
     */
    static refreshAccessToken(apiConfig) {
        return __awaiter(this, void 0, void 0, function* () {
            let data = yield kits_1.HttpKit.getHttpDelegate.httpPost(this.getComponentTokenUrl, JSON.stringify({
                component_appid: apiConfig.getAppId,
                component_appsecret: apiConfig.getAppScrect,
                component_verify_ticket: apiConfig.getTicket
            }));
            if (data) {
                data = JSON.stringify(data);
                let accessToken = new AccessToken_1.AccessToken(data, AccessToken_1.AccessTokenType.COMPONENT_TOKEN);
                let cache = ApiConfigKit_1.ApiConfigKit.getCache;
                cache.set(apiConfig.getAppId, accessToken.getCacheJson);
                return accessToken;
            }
            else {
                throw new Error('获取 accessToken 异常');
            }
        });
    }
}
exports.OpenComponentAccessTokenApi = OpenComponentAccessTokenApi;
OpenComponentAccessTokenApi.getComponentTokenUrl = 'https://api.weixin.qq.com/cgi-bin/component/api_component_token';
