// @tnwx/wxcp/dist/QyDepartmentApi.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
const util = require("util");
const kits_1 = require("@tnwx/kits");
const accesstoken_1 = require("@tnwx/accesstoken");
/**
 * @author Javen
 * @copyright javendev@126.com
 * @description 部门管理相关接口
 */
class QyDepartmentApi {
    /**
     * 创建部门
     * @param name 部门名称
     * @param parentId 父部门id，32位整型
     * @param nameEn 英文名称
     * @param order 在父部门中的次序值
     * @param id 部门id，32位整型，指定时必须大于1
     * @param accessToken {AccessToken}
     */
    static create(name, parentId, nameEn, order, id, accessToken) {
        return __awaiter(this, void 0, void 0, function* () {
            if (!accessToken) {
                accessToken = yield accesstoken_1.QyAccessTokenApi.getAccessToken();
            }
            let url = util.format(this.createUrl, accessToken.getAccessToken);
            return kits_1.HttpKit.getHttpDelegate.httpPost(url, JSON.stringify({
                name: name,
                name_en: nameEn,
                parentid: parentId,
                order: order,
                id: id
            }));
        });
    }
    /**
     * 更新部门
     * @param id 部门id
     * @param name 部门名称
     * @param parentId 父部门id
     * @param nameEn 英文名称
     * @param order 在父部门中的次序值
     * @param accessToken {AccessToken}
     */
    static update(id, name, parentId, nameEn, order, accessToken) {
        return __awaiter(this, void 0, void 0, function* () {
            if (!accessToken) {
                accessToken = yield accesstoken_1.QyAccessTokenApi.getAccessToken();
            }
            let url = util.format(this.updateUrl, accessToken.getAccessToken);
            return kits_1.HttpKit.getHttpDelegate.httpPost(url, JSON.stringify({
                id: id,
                name: name,
                name_en: nameEn,
                parentid: parentId,
                order: order
            }));
        });
    }
    /**
     * 删除部门
     * @param id 部门id
     * @param accessToken {AccessToken}
     */
    static delete(id, accessToken) {
        return __awaiter(this, void 0, void 0, function* () {
            if (!accessToken) {
                accessToken = yield accesstoken_1.QyAccessTokenApi.getAccessToken();
            }
            let url = util.format(this.deleteUrl, accessToken.getAccessToken, id);
            return kits_1.HttpKit.getHttpDelegate.httpGet(url);
        });
    }
    /**
     * 获取部门列表
     * @param id 部门id
     * @param accessToken {AccessToken}
     */
    static get(id, accessToken) {
        return __awaiter(this, void 0, void 0, function* () {
            if (!accessToken) {
                accessToken = yield accesstoken_1.QyAccessTokenApi.getAccessToken();
            }
            let url = util.format(this.getUrl, accessToken.getAccessToken, id);
            return kits_1.HttpKit.getHttpDelegate.httpGet(url);
        });
    }
}
exports.QyDepartmentApi = QyDepartmentApi;
QyDepartmentApi.createUrl = 'https://qyapi.weixin.qq.com/cgi-bin/department/create?access_token=%s';
QyDepartmentApi.updateUrl = 'https://qyapi.weixin.qq.com/cgi-bin/department/update?access_token=%s';
QyDepartmentApi.deleteUrl = 'https://qyapi.weixin.qq.com/cgi-bin/department/delete?access_token=%s&id=%s';
QyDepartmentApi.getUrl = 'https://qyapi.weixin.qq.com/cgi-bin/department/list?access_token=%s&id=%s';
