// @tnwx/commons/dist/entity/menu/Menu.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
/**
 * @author Javen
 * @copyright javendev@126.com
 * @description 自定义菜单
 */
class Menu {
    get getButton() {
        return this.button;
    }
    set setButton(button) {
        this.button = button;
    }
    get getMatchrule() {
        return this.matchrule;
    }
    set setMatchrule(matchrule) {
        this.matchrule = matchrule;
    }
}
exports.Menu = Menu;
class Button {
    constructor(name, type) {
        this.name = name || '';
        this.type = type || '';
    }
    get getName() {
        return this.name;
    }
    set setName(name) {
        this.name = name;
    }
    get getType() {
        return this.type;
    }
    set setType(type) {
        this.type = type;
    }
}
exports.Button = Button;
class ClickButton extends Button {
    constructor(name, type, key) {
        super(name, type);
        this.key = key || '';
    }
    get getKey() {
        return this.key;
    }
    set setKey(key) {
        this.key = key;
    }
}
exports.ClickButton = ClickButton;
class ComButton extends Button {
    constructor(name, type, sub_button) {
        super(name, type);
        this.sub_button = sub_button || [];
    }
    get getSubButton() {
        return this.sub_button;
    }
    set setSubButton(sub_button) {
        this.sub_button = sub_button;
    }
}
exports.ComButton = ComButton;
class MediaButton extends Button {
    constructor(name, type, media_id) {
        super(name, type);
        this.media_id = media_id || '';
    }
    get getMediaId() {
        return this.media_id;
    }
    set setMediaId(media_id) {
        this.media_id = media_id;
    }
}
exports.MediaButton = MediaButton;
class ViewButton extends Button {
    constructor(name, type, url) {
        super(name, type);
        this.url = url || '';
    }
    get getUrl() {
        return this.url;
    }
    set setUrl(url) {
        this.url = url;
    }
}
exports.ViewButton = ViewButton;
class Matchrule {
    get getTagId() {
        return this.tag_id;
    }
    set setTagId(tag_id) {
        this.tag_id = tag_id;
    }
    get getSex() {
        return this.sex;
    }
    set setSex(sex) {
        this.sex = sex;
    }
    get getCountry() {
        return this.country;
    }
    set setCountry(country) {
        this.country = country;
    }
    get getProvince() {
        return this.province;
    }
    set setProvince(province) {
        this.province = province;
    }
    get getCity() {
        return this.city;
    }
    set setCity(city) {
        this.city = city;
    }
    get getClientPlatformType() {
        return this.client_platform_type;
    }
    set setClientPlatformType(client_platform_type) {
        this.client_platform_type = client_platform_type;
    }
    get getLanguage() {
        return this.language;
    }
    set setLanguage(language) {
        this.language = language;
    }
}
exports.Matchrule = Matchrule;
